<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>concordance-nf - Andersen Lab Dry Guide</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "concordance-nf";
        var mkdocs_page_input_path = "pipelines/pipeline-concordance.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Andersen Lab Dry Guide
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Home</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../..">Overview</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Knowledge Base</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/best_practices/">Best Practices</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/r/">R</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/shiny/">R Shiny</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/bash/">Command Line</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/github/">Git and Github</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Rockfish</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/VAST/">Data on Rockfish</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quest</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/b1059/">Data on b1059</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Nextflow Pipelines</span></p>
              <ul class="current">
                  <li class="toctree-l1 current"><a class="reference internal current" >WI Sequencing pipelines</a>
    <ul class="current">
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                </li>
                <li class="toctree-l2 current"><a class="reference internal current" href="#">concordance-nf</a>
    <ul class="current">
    </ul>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" >Other pipelines</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../nf-GCPconfig/">GCP</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other things</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/caendr/">CaeNDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/backup/">Backup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sra/">SRA</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/cloud/">Cloud</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sharing-globus/">Globus file sharing</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/writing-nextflow/">Writing Nextflow workflows</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/labsite/">Andersen Labsite</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Andersen Lab Dry Guide</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Nextflow Pipelines</li>
          <li class="breadcrumb-item">WI Sequencing pipelines</li>
      <li class="breadcrumb-item active">concordance-nf</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipelines/pipeline-concordance.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="concordance-nf">concordance-nf<a class="headerlink" href="#concordance-nf" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#concordance-nf">concordance-nf</a></li>
<li><a href="#pipeline_overview">Pipeline overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a><ul>
<li><a href="#relevant_docker_images">Relevant Docker Images</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#testing_on_rockfish">Testing on Rockfish</a></li>
<li><a href="#running_on_rockfish">Running on Rockfish</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#-profile">-profile</a></li>
<li><a href="#--bam_coverage">--bam_coverage</a></li>
<li><a href="#--vcf">--vcf</a></li>
<li><a href="#--species_optional">--species (optional)</a></li>
<li><a href="#--concordance_cutoff_optional">--concordance_cutoff (optional)</a></li>
<li><a href="#--cores_optional">--cores (optional)</a></li>
<li><a href="#--out_optional">--out (optional)</a></li>
</ul>
</li>
<li><a href="#output">Output</a></li>
</ul>
</div>
<p>The <a href="https://github.com/AndersenLab/concordance-nf"><code>concordance-nf</code></a> pipeline is used to detect sample swaps and determine which wild isolate strains should be grouped together as an isotype. </p>
<p>To determine which strains belong to the same isotype we use two criteria. First we look at the strains that group together with a concordance threshold of 99.95%. Generally this will group most isotypes without issue. However, it is possible that you will run into cases where the grouping is not clean. For example, strain A groups with B, B groups with C, but C does not group with A. In these cases you must examine the data closely to identify why strains are incompletely grouping. Our second criteria we use to group isotypes may address these types of groupings.</p>
<p>The second criteria that we use to group isotypes regards looking for regional differences among strains. If two strains are similar but possess a region of their genome (binned at 1 Mb) that differs by more than 2% then we will separate them out into their own isotypes.</p>
<p><strong>The process of grouping isotypes is very hand-on. This pipeline will help process the data but you must carefully review the output and investigate closely.</strong></p>
<h1 id="pipeline_overview">Pipeline overview<a class="headerlink" href="#pipeline_overview" title="Permanent link">&para;</a></h1>
<pre><code>
    ┌─┐┌─┐┌┐┌┌─┐┌─┐┬─┐┌┬┐┌─┐┌┐┌┌─┐┌─┐  ┌┐┌┌─┐
    │  │ │││││  │ │├┬┘ ││├─┤││││  ├┤───│││├┤ 
    └─┘└─┘┘└┘└─┘└─┘┴└──┴┘┴ ┴┘└┘└─┘└─┘  ┘└┘└  

    parameters              description                                                             Set/Default
    ==========              ===========                                                             =======

    --debug                 Set to 'true' to test                                                   false
    --cores                 Regular job cores                                                       4
    --out                   Directory to output results                                             concordance-{date}
    --vcf                   Hard filtered vcf                                                       null
    --bam_coverage          Table with &quot;strain&quot; and &quot;coverage&quot; as header                            null
    --info_sheet            Strain sheet containing exisiting isotype assignment                    null
    --species               'c_elegans' will check for npr1. All other values will skip this        null
    --concordance_cutoff    Cutoff of concordance value to count two strains as same isotype        0.9995

</code></pre>
<p><img alt="concordance-pipeline" src="../../img/concordance-nf.drawio.svg" /></p>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 23+. On Rockfish, you can access this version by loading the <code>nf23_env</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda
source activate /data/eande106/software/conda_envs/nf23_env
</code></pre>
<h3 id="relevant_docker_images">Relevant Docker Images<a class="headerlink" href="#relevant_docker_images" title="Permanent link">&para;</a></h3>
<p><em>Note: Before 20220301, this pipeline was run using existing conda environments on QUEST. However, these have since been migrated to docker imgaes to allow for better control and reproducibility across platforms. If you need to access the conda version, you can always run an old commit with <code>nextflow run andersenlab/concordance-nf -r 20220216-Release</code></em></p>
<ul>
<li><code>andersenlab/concordance</code> (<a href="https://hub.docker.com/r/andersenlab/concordance">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/concordance.Dockerfile</code> or <code>.github/workflows/build_docker.yml</code> GitHub actions will create a new docker image and push if successful</li>
</ul>
<p>Make sure that you add the following code to your <code>~/.bash_profile</code>. This line makes sure that any singularity images you download will go to a shared location on <code>/vast/eande106</code> for other users to take advantage of (without them also having to download the same image).</p>
<pre><code># add singularity cache
export SINGULARITY_CACHEDIR='/vast/eande106/singularity/'
</code></pre>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you need to work with the docker container, you will need to create an interactive session as singularity can't be run on Rockfish login nodes.</p>
<p><code>interact -n1 -pexpress
module load singularity
singularity shell [--bind local_dir:container_dir] /vast/eande106/singularity/&lt;image_name&gt;</code></p>
</div>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../../rockfish/rf-nextflow/">Nextflow</a> page.</em></p>
<h2 id="testing_on_rockfish">Testing on Rockfish<a class="headerlink" href="#testing_on_rockfish" title="Permanent link">&para;</a></h2>
<p><em>This command uses a test dataset</em></p>
<pre><code>nextflow run -latest andersenlab/concordance-nf --debug
</code></pre>
<h2 id="running_on_rockfish">Running on Rockfish<a class="headerlink" href="#running_on_rockfish" title="Permanent link">&para;</a></h2>
<p>You should run this in a screen or tmux session.</p>
<pre><code>nextflow run -latest andersenlab/concordance-nf --vcf=a.vcf.gz --bam_coverage=mqc_mosdepth-coverage-per-contig_1.txt
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<h2 id="-profile">-profile<a class="headerlink" href="#-profile" title="Permanent link">&para;</a></h2>
<p>There are three configuration profiles for this pipeline.</p>
<ul>
<li><code>rockfish</code> - Used for running on Rockfish (default).</li>
<li><code>quest</code>    - Used for running on Quest.</li>
<li><code>local</code>    - Used for local development.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you forget to add a <code>-profile</code>, the <code>rockfish</code> profile will be chosen as default</p>
</div>
<h2 id="--bam_coverage">--bam_coverage<a class="headerlink" href="#--bam_coverage" title="Permanent link">&para;</a></h2>
<p>The sample sheet to use. This is generally the same sample sheet used for <code>wi-gatk</code>. The sample sheet should look like this:</p>
<p><img alt="Sample_sheet" src="../../img/concordance_sample_sheet.png" /></p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>It is essential that you always use the pipelines and scripts to generate this sample sheet and <strong>NEVER</strong> manually. There are lots of strains and we want to make sure the entire process can be reproduced.</p>
</div>
<h2 id="--vcf">--vcf<a class="headerlink" href="#--vcf" title="Permanent link">&para;</a></h2>
<p>The hard-filtered VCF output from <code>wi-gatk</code>.</p>
<h2 id="--species_optional">--species (optional)<a class="headerlink" href="#--species_optional" title="Permanent link">&para;</a></h2>
<p>Common options include 'c_elegans', 'c_briggsae', and 'c_tropicalis'.</p>
<h2 id="--concordance_cutoff_optional">--concordance_cutoff (optional)<a class="headerlink" href="#--concordance_cutoff_optional" title="Permanent link">&para;</a></h2>
<p>Cutoff to use to determine isotype groups. Default is 0.9995.</p>
<h2 id="--cores_optional">--cores (optional)<a class="headerlink" href="#--cores_optional" title="Permanent link">&para;</a></h2>
<p>The number of cores to use during alignments and variant calling.</p>
<h2 id="--out_optional">--out (optional)<a class="headerlink" href="#--out_optional" title="Permanent link">&para;</a></h2>
<p>A directory in which to output results. By default it will be <code>concordance-YYYYMMDD</code> where YYYYMMDD is todays date.</p>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<pre><code>
├── concordance
    ├── gtcheck.tsv
    ├── isotype_count.txt
    ├── isotype_groups.tsv
    ├── problem_strains.tsv
    ├── WI_metadata.tsv
    ├── concordance.pdf/png
    ├── xconcordance.pdf/png
    └── pairwise
        └── within_group
                └── {isotype_group}.{isotype}.{strain1}_{strain2}.png


</code></pre>
<ul>
<li><strong>concordance.png/pdf</strong> - An image showing the distribution of pairwise concordances across all strains. The cutoff is at 99.9% above which pairs are considered to be in the same isotype unless issues arise.</li>
</ul>
<p><img alt="concordance" src="../../img/concordance.png" /></p>
<ul>
<li><strong>xconcordance.png/pdf</strong> - A close up view of the concordances showing more detail. </li>
</ul>
<p><img alt="concordance above 99" src="../../img/concordance_above_99.png" /></p>
<ul>
<li><strong>isotype_groups.tsv</strong> - <em>This is the one of the most important output files</em>. It illustrates the isotypes identified for each strain and identifies potential issues.</li>
</ul>
<p>A file with the following structure:</p>
<p><img alt="pairwise" src="../../img/pairwise_output_example.png" /></p>
<ul>
<li><strong>group</strong> - A number used to group strains (in each row) into an isotype automatically. This number should be unique with the isotype column (e.g. 1--&gt; AB1, 112 --&gt; CB4858, BRC20067 --&gt; 175). The number can change between analyses.</li>
<li><strong>strain</strong> - the strain</li>
<li><strong>isotype</strong> - the currently assigned isotype for a strain taken from the <code>WI Strain Info</code> spreadsheet. When new strains are added this is blank. </li>
<li><strong>latitude</strong></li>
<li><strong>longitude</strong></li>
<li><strong>coverage</strong> - Depth of coverage for strain.</li>
<li><strong>unique_isotypes_per_group</strong> - Number of unique isotypes when grouping by the group column. This should be 1. If it is more than 1, it indicates that multiple isotypes were assigned to a grouping and that a previously assigned isotype is now being called into question.</li>
<li><strong>unique_groups_per_isotype</strong> Number of unique groups assigned to an isotype. This should be 1. If it is higher than 1, it indicates that a strain is concordant with strains in two different isotypes (including blanks). If it is equal to 2 and contains blanks in the isotype column it likely means that an isotype should be assigned to that strain.</li>
<li><strong>strain_in_multiple_isotypes</strong> - Indicates that a strain is falling into multiple isotypes (a problem!).</li>
<li><strong>location_issue</strong> - Indicates a location issue. This occurs when strains fall into an isotype but are located far away from one another. Some are known issues and can be ignored.</li>
<li><strong>strain_conflict</strong> - <code>TRUE</code> if any issue is present that should be investigated.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This file might change as you manually adjust the concordance cutoff for each run</p>
</div>
<ul>
<li>
<p><strong>gtcheck.tsv</strong> - the <strong>other most important file</strong>. File produced using <code>bcftools gtcheck</code>; Raw genotype differences between strains. This file is used in manual inspection of the isotype groups</p>
</li>
<li>
<p><strong>isotype_count.txt</strong> - Gives a count of the number of isotypes identified.</p>
</li>
<li>
<p>concordance/pairwise/ (directory)</p>
</li>
</ul>
<p>Contains images showing locations where regional discordance occurs among strains classified as being the isotype. You must look through all these images to ensure there are no strains being grouped that have regions with significant differences (&gt; 2%). The image below illustrates an example of this. ED3049 and ED3046 are highly similar (&gt; 99.9%). However, they differ in a region on the right arm of chromosome II. We believe this was enough reason to consider them separate isotypes.</p>
<p><img alt="pairwise example" src="../../img/194.ED3049.ED3046_ED3049.png" /></p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../pipeline-wiGATK/" class="btn btn-neutral float-left" title="wi-gatk"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../pipeline-postGATK/" class="btn btn-neutral float-right" title="post-gatk-nf">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-wiGATK/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../pipeline-postGATK/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/main.js"></script>
      <script src="../../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
