<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>post-gatk-nf - Andersen Lab Dry Guide</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "post-gatk-nf";
        var mkdocs_page_input_path = "pipelines/pipeline-postGATK.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Andersen Lab Dry Guide
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Home</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../..">Overview</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Knowledge Base</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/best_practices/">Best Practices</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/r/">R</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/shiny/">R Shiny</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/bash/">Command Line</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/github/">Git and Github</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Rockfish</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/VAST/">Data on Rockfish</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quest</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/b1059/">Data on b1059</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Nextflow Pipelines</span></p>
              <ul class="current">
                  <li class="toctree-l1 current"><a class="reference internal current" >WI Sequencing pipelines</a>
    <ul class="current">
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                </li>
                <li class="toctree-l2 current"><a class="reference internal current" href="#">post-gatk-nf</a>
    <ul class="current">
    </ul>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" >Other pipelines</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../nf-GCPconfig/">GCP</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other things</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/caendr/">CeNDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/backup/">Backup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sra/">SRA</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/cloud/">Cloud</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sharing-globus/">Globus file sharing</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/writing-nextflow/">Writing Nextflow workflows</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/labsite/">Andersen Labsite</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Andersen Lab Dry Guide</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Nextflow Pipelines</li>
          <li class="breadcrumb-item">WI Sequencing pipelines</li>
      <li class="breadcrumb-item active">post-gatk-nf</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipelines/pipeline-postGATK.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="post-gatk-nf">post-gatk-nf<a class="headerlink" href="#post-gatk-nf" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#post-gatk-nf">post-gatk-nf</a></li>
<li><a href="#pipeline_overview">Pipeline overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a><ul>
<li><a href="#relevant_docker_images">Relevant Docker Images</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#testing_on_rockfish">Testing on Rockfish</a></li>
<li><a href="#running_on_rockfish">Running on Rockfish</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#-profile">-profile</a></li>
<li><a href="#--debug">--debug</a><ul>
<li><a href="#debugging_for_pca">Debugging for PCA:</a></li>
</ul>
</li>
<li><a href="#--sample_sheet">--sample_sheet</a></li>
<li><a href="#--vcf_folder">--vcf_folder</a></li>
<li><a href="#--species_optional">--species (optional)</a></li>
<li><a href="#pca">PCA</a></li>
<li><a href="#--postgatk_default_true">--postgatk (default: true)</a></li>
<li><a href="#--pca_default_false">--pca (default: false)</a></li>
<li><a href="#--snv_vcf_pca">--snv_vcf (pca)</a></li>
<li><a href="#--pops_pca">--pops (pca)</a></li>
<li><a href="#--eigen_ld_pca">--eigen_ld (pca)</a></li>
<li><a href="#--anc_pca">--anc (pca)</a></li>
<li><a href="#--output_optional">--output (optional)</a></li>
</ul>
</li>
<li><a href="#output">Output</a></li>
<li><a href="#data_storage">Data storage</a><ul>
<li><a href="#cleanup">Cleanup</a></li>
<li><a href="#updating_caendr">Updating CaeNDR</a></li>
<li><a href="#updating_nemascan">Updating NemaScan</a></li>
</ul>
</li>
</ul>
</div>
<p>The <a href="https://github.com/AndersenLab/post-gatk-nf">post-gatk-nf</a> pipeline performs population genetics analyses (such as identifying shared haplotypes and divergent regions) at the isotype level. The VCFs output from this pipeline are used within the lab and also released to the world via CaeNDR.</p>
<p>This page details how to run the pipeline.</p>
<h1 id="pipeline_overview">Pipeline overview<a class="headerlink" href="#pipeline_overview" title="Permanent link">&para;</a></h1>
<pre><code>

      * * * *                    **           * * * *    * * *    * * * *    *   *                         *
     *       *                * * * * *     *        *  *     *      *       *  *                         * *
    *        *                   **         *           *     *      *       * *                         * *
   *        *   * * * * * *      **    ***  *           * * * *      *       * *      ***      *          *
  * * * * *    *   * *   * *     **         *    * * *  *     *      *       *  *             * * *      *
 *            *     *   *   *   *  *        *        *  *     *      *       *   *           *     *    *   *
*              * * *   * * * * *    *        * * * * *  *     *      *       *    *         *      * * * * *  
                                                                                                      **
                                                                                                     * * 
                                                                                                    *  *
                                                                                                   *  *
                                                                                                    *

    parameters              description                                            Set/Default
    ==========              ===========                                            ========================
    --debug                 Use --debug to indicate debug mode                     (optional)
    --vcf_folder            Folder to hard and soft filtered vcf                   (required)
    --sample_sheet          TSV with column iso-ref strain, bam, bai (no header)   (required)
    --species               Species: 'c_elegans', 'c_tropicalis' or 'c_briggsae'   c_elegans
    --output                Output folder name.                                    popgen-date (in current folder)
    --postgatk              Run post-GATK steps                                    true
    --pca                   Run PCA analysis                                       false


</code></pre>
<p><img alt="Pipeline-overview" src="../../img/drawio/post-gatk.drawio.svg" /></p>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 23+. On Rockfish, you can access this version by loading the <code>nf23_env</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda
source activate /data/eande106/software/conda_envs/nf23_env
</code></pre>
<h3 id="relevant_docker_images">Relevant Docker Images<a class="headerlink" href="#relevant_docker_images" title="Permanent link">&para;</a></h3>
<p><em>Note: Before 20220301, this pipeline was run using existing conda environments on QUEST. However, these have since been migrated to docker imgaes to allow for better control and reproducibility across platforms. If you need to access the conda version, you can always run an old commit with <code>nextflow run andersenlab/post-gatk-nf -r 20220216-Release</code></em></p>
<ul>
<li><code>andersenlab/postgatk</code> (<a href="https://hub.docker.com/r/andersenlab/postgatk">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/postgatk.Dockerfile</code> or <code>.github/workflows/build_postgatk_docker.yml</code> GitHub actions will create a new docker image and push if successful</li>
<li><code>andersenlab/tree</code> (<a href="https://hub.docker.com/r/andersenlab/tree">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/tree.Dockerfile</code> or <code>.github/workflows/build_tree_docker.yml</code> GitHub actions will create a new docker image and push if successful</li>
<li><code>andersenlab/pca</code> (<a href="https://hub.docker.com/r/andersenlab/pca">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/pca.Dockerfile</code> or <code>.github/workflows/build_pca_docker.yml</code> GitHub actions will create a new docker image and push if successful</li>
<li><code>andersenlab/r_packages</code> (<a href="https://hub.docker.com/r/andersenlab/r_packages">link</a>): Docker image is created manually, code can be found in the <a href="https://github.com/AndersenLab/dockerfile/tree/master/r_packages">dockerfile</a> repo.</li>
</ul>
<p>Make sure that you add the following code to your <code>~/.bash_profile</code>. This line makes sure that any singularity images you download will go to a shared location on <code>/vast/eande106</code> for other users to take advantage of (without them also having to download the same image).</p>
<pre><code># add singularity cache
export SINGULARITY_CACHEDIR='/vast/eande106/singularity/'
</code></pre>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you need to work with the docker container, you will need to create an interactive session as singularity can't be run on Rockfish login nodes.</p>
<p><code>interact -n1 -pexpress
module load singularity
singularity shell [--bind local_dir:container_dir] /vast/eande106/singularity/&lt;image_name&gt;</code></p>
</div>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../../rockfish/rf-nextflow/">Nextflow</a> page.</em></p>
<h2 id="testing_on_rockfish">Testing on Rockfish<a class="headerlink" href="#testing_on_rockfish" title="Permanent link">&para;</a></h2>
<p><em>This command uses a test dataset</em></p>
<pre><code>nextflow run -latest andersenlab/post-gatk-nf --debug
</code></pre>
<h2 id="running_on_rockfish">Running on Rockfish<a class="headerlink" href="#running_on_rockfish" title="Permanent link">&para;</a></h2>
<p>You should run this in a screen or tmux session.</p>
<pre><code>nextflow run -latest andersenlab/post-gatk-nf --vcf &lt;path_to_vcf&gt; --sample_sheet &lt;path_to_sample_sheet&gt;
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<h2 id="-profile">-profile<a class="headerlink" href="#-profile" title="Permanent link">&para;</a></h2>
<p>There are three configuration profiles for this pipeline.</p>
<ul>
<li><code>rockfish</code> - Used for running on Rockfish (default).</li>
<li><code>quest</code>    - Used for running on Quest.</li>
<li><code>local</code>    - Used for local development.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you forget to add a <code>-profile</code>, the <code>rockfish</code> profile will be chosen as default</p>
</div>
<h2 id="--debug">--debug<a class="headerlink" href="#--debug" title="Permanent link">&para;</a></h2>
<p>You should use <code>--debug</code> for testing/debugging purposes. This will run the debug test set (located in the <code>test_data</code> folder).</p>
<p>For example:</p>
<pre><code>nextflow run andersenlab/post-gatk-nf --debug
</code></pre>
<p>Using <code>--debug</code> will automatically set the sample sheet to <code>test_data/sample_sheet.tsv</code></p>
<h3 id="debugging_for_pca">Debugging for PCA:<a class="headerlink" href="#debugging_for_pca" title="Permanent link">&para;</a></h3>
<p>You can debug the PCA pipeline with the following data/command:</p>
<pre><code>nextflow run -latest andersenlab/post-gatk-nf --vcf ./test_data/WI.20220404.hard-filter.vcf.gz --species c_elegans --sample_sheet ./test_data/sample_sheet_2.tsv --eigen_ld 0.8,0.6 --anc XZ2019 --pca -resume
</code></pre>
<h2 id="--sample_sheet">--sample_sheet<a class="headerlink" href="#--sample_sheet" title="Permanent link">&para;</a></h2>
<p>A custom sample sheet can be specified using <code>--sample_sheet</code>. The <code>sample sheet</code> is generated from the sample sheet used as input for <a href="https://github.com/AndersenLab/wi-gatk"><code>wi-gatk</code></a> with only columns for strain, bam, and bai subsetted. <strong>Make sure to remove any strains that you do not want to include in this analysis.</strong> (<em>i.e. subset to keep only ISOTYPE strains</em>)</p>
<p>Remember that in <code>--debug</code> mode the pipeline will use the sample sheet located in <code>test_data/sample_sheet.tsv</code>.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>There is no header for the sample sheet!</p>
</div>
<p>The <code>sample sheet</code> has the following columns:</p>
<ul>
<li><strong>strain</strong> - the name of the strain</li>
<li><strong>bam</strong> - name of the bam alignment file</li>
<li><strong>bai</strong> - name of the bam alignment index file</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>As of 20210501, bam and bam.bai files for all strains of a particular species can be found in one singular location: <code>/vast/eande106/data/{species}/WI/alignments/</code> so there is no longer need to provide the location of the bam files.</p>
</div>
<h2 id="--vcf_folder">--vcf_folder<a class="headerlink" href="#--vcf_folder" title="Permanent link">&para;</a></h2>
<p>Path to the <strong>folder</strong> containing both the hard-filtered and soft-filtered vcf outputs from <a href="https://github.com/AndersenLab/wi-gatk"><code>wi-gatk</code></a>. VCF should contain <strong>ALL</strong> strains, the first step will be to subset isotype reference strains for further analysis.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This should be the <strong>path to the folder</strong>, we want to isotype-subset both hard and soft filtered VCFs. For example: <code>--vcf_folder /vast/eande106/projects/Katie/wi-gatk/WI-20210121/variation/</code> or <code>--vcf_folder /vast/eande106/data/c_elegans/WI/variation/20210121/vcf/</code></p>
</div>
<h2 id="--species_optional">--species (optional)<a class="headerlink" href="#--species_optional" title="Permanent link">&para;</a></h2>
<p><strong>default</strong> = c_elegans</p>
<p>Options: c_elegans, c_briggsae, or c_tropicalis</p>
<h2 id="pca">PCA<a class="headerlink" href="#pca" title="Permanent link">&para;</a></h2>
<p>The PCA steps can be run either with the full pipeline or independently. To run only PCA use <code>-- pca true --postgatk false</code> </p>
<p>The input VCF is filtered to bi-alleleic snps with no missing genotypes. A LD filtering threshold is required and LD filtering is performed using plink. You can also filter for singletons by specifying the <code>--singletons</code> </p>
<p>PCA is performed using smartPCA. Parameters to control outlier threshold or removal iterations are desribed below.</p>
<h2 id="--postgatk_default_true">--postgatk (default: true)<a class="headerlink" href="#--postgatk_default_true" title="Permanent link">&para;</a></h2>
<p>Set to false to skip post-gatk steps</p>
<h2 id="--pca_default_false">--pca (default: false)<a class="headerlink" href="#--pca_default_false" title="Permanent link">&para;</a></h2>
<p>Set to true to run PCA analysis</p>
<h2 id="--snv_vcf_pca">--snv_vcf (pca)<a class="headerlink" href="#--snv_vcf_pca" title="Permanent link">&para;</a></h2>
<p>File path to SNV-filtered VCF</p>
<h2 id="--pops_pca">--pops (pca)<a class="headerlink" href="#--pops_pca" title="Permanent link">&para;</a></h2>
<p>Strain list to filter VCF for PCA analysis. No header:</p>
<table>
<thead>
<tr>
<th>AB1</th>
</tr>
</thead>
<tbody>
<tr>
<td>CB4856</td>
</tr>
<tr>
<td>ECA788</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you run the standard profile with pca this file will be automatically generated to include all isotypes.</p>
</div>
<h2 id="--eigen_ld_pca">--eigen_ld (pca)<a class="headerlink" href="#--eigen_ld_pca" title="Permanent link">&para;</a></h2>
<p>LD thresholds to test for PCA. Can provide multiple with <code>--eigen_ld 0.8,0.6,0.4</code></p>
<h2 id="--anc_pca">--anc (pca)<a class="headerlink" href="#--anc_pca" title="Permanent link">&para;</a></h2>
<p>Ancestor strain to use for PCA. </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Make sure this strain is in your VCF</p>
</div>
<h2 id="--output_optional">--output (optional)<a class="headerlink" href="#--output_optional" title="Permanent link">&para;</a></h2>
<p><strong>default</strong> - <code>popgen-YYYYMMDD</code></p>
<p>A directory in which to output results. If you have set <code>--debug</code>, the default output directory will be <code>popgen-YYYYMMDD-debug</code>.</p>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<pre><code>├── ANNOTATE_VCF
│   ├── ANC.bed.gz
│   ├── ANC.bed.gz.tbi
│   ├── Ce330_annotated.vcf.gz
|   └── Ce330_annotated.vcf.tbi
├── EIGESTRAT
│   └── LD_{eigen_ld}
│       ├── INPUT_FILES
│       │   └── *
│       ├── OUTLIER_REMOVAL
│       │   ├── eigenstrat_outliers_removed_relatedness
│       │   ├── eigenstrat_outliers_removed_relatedness.id
│       │   ├── eigenstrat_outliers_removed.evac
│       │   ├── eigenstrat_outliers_removed.eval
│       │   ├── logfile_outlier.txt
│       │   └── TracyWidom_statistics_outlier_removal.tsv
│       └── NO_REMOVAL
│           └── same as outlier_removal
├── pca_report.html
├── divergent_regions
│   ├── Mask_DF
│   │   └── [strain]_Mask_DF.tsv
|   └── divergent_regions_strain.bed
├── haplotype
│   ├── haplotype_length.pdf
│   ├── sweep_summary.tsv
│   ├── max_haplotype_genome_wide.pdf
│   ├── haplotype.pdf
│   ├── haplotype.tsv
│   ├── [chr].ibd
│   └── haplotype_plot_df.Rda
├── tree
│   ├── WI.{date}.hard-filter.isotype.min4.tree
│   ├── WI.{date}.hard-filter.isotype.min4.tree.pdf
│   ├── WI.{date}.hard-filter.min4.tree
│   └── WI.{date}.hard-filter.min4.tree.pdf
├── NemaScan
│   ├── strain_isotype_lookup.tsv
│   ├── div_isotype_list.txt
│   ├── haplotype_df_isotype.bed
│   ├── divergent_bins.bed
│   └── divergent_df_isotype.bed
└── variation
    ├── WI.{date}.small.hard-filter.isotype.vcf.gz
    ├── WI.{date}.small.hard-filter.isotype.vcf.gz.tbi
    ├── WI.{date}.hard-filter.isotype.SNV.vcf.gz
    ├── WI.{date}.hard-filter.isotype.SNV.vcf.gz.tbi
    ├── WI.{date}.soft-filter.isotype.vcf.gz
    ├── WI.{date}.soft-filter.isotype.vcf.gz.tbi
    ├── WI.{date}.hard-filter.isotype.vcf.gz
    └── WI.{date}.hard-filter.isotype.vcf.gz.tbi

</code></pre>
<h1 id="data_storage">Data storage<a class="headerlink" href="#data_storage" title="Permanent link">&para;</a></h1>
<h2 id="cleanup">Cleanup<a class="headerlink" href="#cleanup" title="Permanent link">&para;</a></h2>
<p>Once the pipeline has complete successfully and you are satisfied with the results, the final data can be moved to their final storage place on Quest accordingly:</p>
<ul>
<li>Everything in the <code>haplotype</code> foder can be moved to <code>/vast/eande106/data/{species}/WI/haplotype/{date}</code></li>
<li>Everything in the <code>divergent_regions</code> folder can be moved to <code>/vast/eande106/data/{species}/WI/divergent_regions/{date}</code></li>
<li>Everything in the <code>tree</code> folder can be moved to <code>/vast/eande106/data/{species}/WI/tree/{date}</code></li>
<li>Everything in the <code>variation</code> folder can be moved to <code>/vast/eande106/data/{species}/WI/variation/{date}/vcf</code></li>
<li>Everything in the <code>NemaScan</code> folder can replace the old verions in NemaScan when ready <code>NemaScan/input_data/{species}/isotypes</code></li>
<li>Everything in the <code>EIGENSTRAT</code> folder can be moved to <code>/vast/eande106/data/{species}/WI/pca/{date}</code></li>
</ul>
<h2 id="updating_caendr">Updating CaeNDR<a class="headerlink" href="#updating_caendr" title="Permanent link">&para;</a></h2>
<p>Check out the <a href="../../other/caendr/">CaeNDR</a> page for more information about updating a new data release for CeNDR.</p>
<h2 id="updating_nemascan">Updating <code>NemaScan</code><a class="headerlink" href="#updating_nemascan" title="Permanent link">&para;</a></h2>
<p>Once a new CaeNDR release is ready, it is important to also update the genome-wide association mapping packages to ensure users can appropriately analyze data from new strains as well as old strains. Here is a list of things that need to be updated:</p>
<ul>
<li>The default vcf should be changed to the newest release date (i.e. from 20200815 to 20210121). Users will still have the option to use an earlier vcf.</li>
<li>Everything in the <code>NemaScan</code> folder can replace the old verions in NemaScan when ready <code>NemaScan/input_data/{species}/isotypes</code></li>
<li>Be sure that the small-vcf is stored in the proper file location to be accessed, both on Rockfish and on GCP for CaeNDR/local users.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Although users will have the option to use an older vcf, the divergent region data will always be pulled from the most recent release. There could be minor changes from release to release. If this is a concern, we could switch to pulling the divergent data directly from <code>/vast/eande106/data</code> instead of including it in the <code>bin/</code> of the mapping pipeline.</p>
</div>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../pipeline-concordance/" class="btn btn-neutral float-left" title="concordance-nf"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../pipeline-impute/" class="btn btn-neutral float-right" title="imputation">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-concordance/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../pipeline-impute/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/main.js"></script>
      <script src="../../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
