<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>R - Andersen Lab Dry Guide</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css" />
  <link href="../css/version-select.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "R";
    var mkdocs_page_input_path = "r.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Andersen Lab Dry Guide</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Knowledge Base</span></p>
                <ul class="current">
                    <li class="toctree-l1"><a class="reference internal" href="../bash/">Bash</a>
                    </li>
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">R</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#andersen_lab_r_packages">Andersen Lab R Packages</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#cegwas2">cegwas2</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#linkagemapping">linkagemapping</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#copasutils">COPASutils</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#easysorter">easysorter</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#easyxpress">easyXpress</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#general_r_resources">General R resources</a>
    </li>
    </ul>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../github/">Git and Github</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Quest</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-intro/">Introduction</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-nextflow/">Nextflow</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-GCPconfig/">GCP</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-conda/">Conda</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-docker/">Docker</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-mount/">Mounting Quest</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Pipelines</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../adding-seq-data/">Adding NILs Sequencing Data</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-cegwas/">cegwas2-nf</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Other things</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../cendr/">CeNDR</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../backup/">Backup</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../cloud/">Cloud</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../labsite/">Andersen Labsite</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Andersen Lab Dry Guide</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Knowledge Base &raquo;</li>
        
      
    
    <li>R</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/r.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="r">R<a class="headerlink" href="#r" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#r">R</a><ul>
<li><a href="#andersen_lab_r_packages">Andersen Lab R Packages</a><ul>
<li><a href="#cegwas2">cegwas2</a></li>
<li><a href="#linkagemapping">linkagemapping</a></li>
<li><a href="#copasutils">COPASutils</a></li>
<li><a href="#easysorter">easysorter</a></li>
<li><a href="#easyxpress">easyXpress</a></li>
</ul>
</li>
<li><a href="#general_r_resources">General R resources</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="andersen_lab_r_packages">Andersen Lab R Packages<a class="headerlink" href="#andersen_lab_r_packages" title="Permanent link">&para;</a></h2>
<p>The Andersen lab maintains several R packages useful for high-throughput data analysis.</p>
<h3 id="cegwas2">cegwas2<a class="headerlink" href="#cegwas2" title="Permanent link">&para;</a></h3>
<p>This package contains a set of functions to process phenotype data, perform GWAS, and perform post-mapping data processing for <em>C. elegans</em>. In 2019, the <code>cegwas2-nf</code> Nextflow pipeline was developed to perform GWA mapping on QUEST using this cegwas2 R package. However, mapping is rarely if never done with cegwas2 in R manually. To learn more about the <code>cegwas2</code> R package, see the <a href="https://github.com/andersenlab/cegwas2">andersenlab/cegwas2</a> repo. For help running a GWA mapping using cegwas2, see <a href="https://github.com/andersenlab/cegwas2-nf">cegwas2-nf</a> or <a href="../pipeline-cegwas/">the dry guide</a></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><code>cegwas2</code> was preceeded by <code>cegwas</code> and will soon be superceeded by <a href="https://github.com/AndersenLab/NemaScan"><code>NemaScan</code></a></p>
</div>
<h3 id="linkagemapping">linkagemapping<a class="headerlink" href="#linkagemapping" title="Permanent link">&para;</a></h3>
<p>This package includes all data and functions necessary to complete a mapping for the phenotype of your choice using the recombinant inbred lines from <a href="https://academic.oup.com/g3journal/article/5/5/911/6025542">Andersen, et al. 2015 (G3)</a>. Included with this package are the cross and map objects for this strain set as well a markers.rds file containing a lookup table for the physical positions of all markers used for mapping. To learn more about linkagemapping including how to install and use the package, check out the <a href="https://github.com/andersenlab/linkagemapping">andersenlab/linkagemapping</a> repo.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Also check out the <a href="https://github.com/andersenlab/linkagemapping-nf">linkagemapping-nf</a> repo for a reproducible Nextflow pipeline for linkage mapping and two-dimensional genome scans (scan2) for one or several traits.</p>
</div>
<h3 id="copasutils">COPASutils<a class="headerlink" href="#copasutils" title="Permanent link">&para;</a></h3>
<p>The R package <code>COPASutils</code> provides a logical workflow for the reading, processing, and visualization of data obtained from the Union Biometrica Complex Object Parametric Analyzer and Sorter (COPAS) or the BioSorter large-particle flow cytometers. Data obtained from these powerful experimental platforms can be unwieldy, leading to difficulties in the ability to process and visualize the data using existing tools. Researchers studying small organisms, such as Caenorhabditis elegans, Anopheles gambiae, and Danio rerio, and using these devices will benefit from this streamlined and extensible R package. <code>COPASutils</code> offers a powerful suite of functions for the rapid processing and analysis of large high-throughput screening data sets. To learn more about COPASutils including how to install and use the package, check out the <a href="https://github.com/andersenlab/copasutils">andersenlab/COPASutils</a> repo and the COPASutils <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0111090">manuscript</a></p>
<h3 id="easysorter">easysorter<a class="headerlink" href="#easysorter" title="Permanent link">&para;</a></h3>
<p>This package is effectively version 2 of the <code>COPASutils</code> package. This package is specialized for use with worms and includes additional functionality on top of that provided by COPASutils, including division of recorded objects by larval stage and the ability to regress out control phenotypes from those recorded in experimental conditions To learn more about easysorter including how to install and use the package, check out the <a href="https://github.com/andersenlab/easysorter">andersenlab/easysorter</a> repo. Here are some of the papers using <code>easysorter</code>:</p>
<ul>
<li>
<p><strong>The first easysorter paper</strong>
A Powerful New Quantitative Genetics Platform, Combining <em>Caenorhabditis elegans</em> High-Throughput Fitness Assays with a Large Collection of Recombinant Strains (<a href="https://academic.oup.com/g3journal/article/5/5/911/6025542">Andersen et al. 2015</a>)</p>
</li>
<li>
<p><strong>The first "V3" easysorter paper</strong>
The Gene <em>scb-1</em> Underlies Variation in <em>Caenorhabditis elegans</em> Chemotherapeutic Responses (<a href="https://academic.oup.com/g3journal/article/10/7/2353/6026329">Evans and Andersen 2020</a>)</p>
</li>
<li>
<p><strong>The first dominance/hemizygosity easysorter paper</strong>
A Novel Gene Underlies Bleomycin-Response Variation in <em>Caenorhabditis elegans</em> (<a href="https://academic.oup.com/genetics/article/212/4/1453/5931413">Brady et al. 2019</a>)</p>
</li>
</ul>
<p><em>Almost every paper published from the lab has used easysorter, for more, check out <a href="https://andersenlab.org/Publications/">our lab papers</a></em></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <code>easysorter</code> package requires <code>COPASutils</code> installation as well.</p>
</div>
<h3 id="easyxpress">easyXpress<a class="headerlink" href="#easyxpress" title="Permanent link">&para;</a></h3>
<p>This package is designed for the reading, processing, and visualization of images obtained from the Molecular Devices ImageExpress Nano Imager, and processed with CellProfiler's WormToolbox. To learn more about easyXpress including how to install and use the package, check out the <a href="https://github.com/andersenlab/easyXpress">andersenlab/easyXpress</a> repo and the <a href="https://www.biorxiv.org/content/10.1101/2021.05.11.443552v1">easyXpress manuscript</a>. </p>
<h2 id="general_r_resources">General R resources<a class="headerlink" href="#general_r_resources" title="Permanent link">&para;</a></h2>
<ul>
<li><a href="https://github.com/katiesevans/nuit_tidyverse">Tidyverse workshop and resources</a></li>
<li><a href="">Andersen Lab R Knowledge base &amp; Cheatsheet</a></li>
<li>Also check out the <code>lab_code</code> slack channel for help/questions!</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../github/" class="btn btn-neutral float-right" title="Git and Github">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../bash/" class="btn btn-neutral" title="Bash"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/Andersenlab/dry-guide/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../bash/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../github/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../js/version-select.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
