<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>wi-gatk - Andersen Lab Dry Guide</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "wi-gatk";
        var mkdocs_page_input_path = "pipeline-wiGATK.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> Andersen Lab Dry Guide
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Knowledge Base</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../best_practices/">Best Practices</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../r/">R</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../shiny/">R Shiny</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../bash/">Command Line</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../github/">Git and Github</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Rockfish</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../rf-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../VAST/">Data on Rockfish</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../rf-nextflow/">Nextflow</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../rf-conda/">Conda</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quest</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../quest-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../b1059/">Data on b1059</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../quest-nextflow/">Nextflow</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../quest-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-GCPconfig/">GCP</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">WI Sequencing pipelines</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">wi-gatk</a>
    <ul class="current">
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other pipelines</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../pipeline-cegwas/">cegwas2-nf</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other things</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../cendr/">CeNDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../backup/">Backup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../cloud/">Cloud</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../sharing-globus/">Globus file sharing</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../labsite/">Andersen Labsite</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">Andersen Lab Dry Guide</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">WI Sequencing pipelines</li>
      <li class="breadcrumb-item active">wi-gatk</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipeline-wiGATK.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="wi-gatk">wi-gatk<a class="headerlink" href="#wi-gatk" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#wi-gatk">wi-gatk</a></li>
<li><a href="#pipeline_overview">Pipeline Overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a><ul>
<li><a href="#relevant_docker_images">Relevant Docker Images</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#profiles">Profiles</a></li>
<li><a href="#running_the_pipeline_locally">Running the pipeline locally</a></li>
<li><a href="#debugging_the_pipeline_on_quest">Debugging the pipeline on Quest</a></li>
<li><a href="#running_the_pipeline_on_quest">Running the pipeline on Quest</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#--sample_sheet">--sample_sheet</a><ul>
<li><a href="#--bam_location_optional">--bam_location (optional)</a></li>
<li><a href="#--species_optional">--species (optional)</a></li>
<li><a href="#--project_optional">--project (optional)</a></li>
<li><a href="#--ws_build_optional">--ws_build (optional)</a></li>
<li><a href="#--reference_optional">--reference (optional)</a></li>
<li><a href="#--mito_name_optional">--mito_name (optional)</a></li>
<li><a href="#--output_optional">--output (optional)</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#output">Output</a></li>
<li><a href="#data_storage">Data Storage</a><ul>
<li><a href="#cleanup">Cleanup</a></li>
</ul>
</li>
</ul>
</div>
<p>The <code>wi-gatk</code> pipeline filters and calls variants from wild isolate sequence data.</p>
<h1 id="pipeline_overview">Pipeline Overview<a class="headerlink" href="#pipeline_overview" title="Permanent link">&para;</a></h1>
<pre><code>
    _______ _______ _______ __  __      _______ _______ 
    |     __|   _   |_     _|  |/  |    |    |  |    ___|
    |    |  |       | |   | |     &lt;     |       |    ___|
    |_______|___|___| |___| |__|\\__|    |__|____|___|    


    parameters                 description                           Set/Default
    ==========                 ===========                           ========================
    --debug                    Use --debug to indicate debug mode    null
    --output                   Release Directory                     WI-{date}
    --sample_sheet             Sample sheet                          null
    --bam_location             Directory of bam files                /projects/b1059/data/{species}/WI/alignments/
    --mito_name                Contig not to polarize hetero sites   MtDNA

    Reference Genome
    --------------- 
    --reference_base           Location of ref genomes               /projects/b1059/data/{species}/genomes/
    --species/project/build    These 4 params form --reference       {species} / {project} / {ws_build}

    Variant Filters         
    ---------------           
    --min_depth                Minimum variant depth                 5
    --qual                     Variant QUAL score                    30
    --strand_odds_ratio        SOR_strand_odds_ratio                 5
    --quality_by_depth         QD_quality_by_depth                   20
    --fisherstrand             FS_fisher_strand                      100
    --high_missing             Max % missing genotypes               0.95
    --high_heterozygosity      Max % max heterozygosity              0.10


</code></pre>
<p><img alt="Pipeline-overview" src="../img/wi-gatk.drawio.svg" /></p>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 20.0+. On QUEST, you can access this version by loading the <code>nf20</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda3.6
source activate /projects/b1059/software/conda_envs/nf20_env
</code></pre>
<p>Alternatively you can update Nextflow by running:</p>
<pre><code>nextflow self-update
</code></pre>
<h3 id="relevant_docker_images">Relevant Docker Images<a class="headerlink" href="#relevant_docker_images" title="Permanent link">&para;</a></h3>
<ul>
<li><code>andersenlab/gatk4</code> (<a href="https://hub.docker.com/r/andersenlab/gatk4">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/gatk4.Dockerfile</code> or <code>.github/workflows/build_docker.yml</code> GitHub actions will create a new docker image and push if successful</li>
<li><code>andersenlab/r_packages</code> (<a href="https://hub.docker.com/r/andersenlab/r_packages">link</a>): Docker image is created manually, code can be found in the <a href="https://github.com/AndersenLab/dockerfile/tree/master/r_packages">dockerfile</a> repo.</li>
</ul>
<p>To access these docker images, first load the <code>singularity</code> module on QUEST.</p>
<pre><code>module load singularity
</code></pre>
<p>Also, make sure that you add the following code to your <code>~/.bash_profile</code>. This line makes sure that any singularity images you download will go to a shared location on <code>b1059</code> for other users to take advantage of (without them also having to download the same image).</p>
<pre><code># add singularity cache
export SINGULARITY_CACHEDIR='/projects/b1059/singularity/'
</code></pre>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../quest-nextflow/">Nextflow</a> page.</em></p>
<h2 id="profiles">Profiles<a class="headerlink" href="#profiles" title="Permanent link">&para;</a></h2>
<p>The <code>nextflow.config</code> file included with this pipeline contains three profiles. These set up the environment for testing local development, testing on Quest, and running the pipeline on Quest.</p>
<ul>
<li><code>local</code> - Used for local development. Uses the docker container.</li>
<li><code>debug</code> - Runs a small subset of available test data. Should complete within a couple of minutes. For testing/diagnosing issues on Quest.</li>
<li><code>quest</code> - Runs the entire dataset.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you forget to add a <code>-profile</code>, the <code>quest</code> profile will be chosen as default</p>
</div>
<h2 id="running_the_pipeline_locally">Running the pipeline locally<a class="headerlink" href="#running_the_pipeline_locally" title="Permanent link">&para;</a></h2>
<p>When running locally, the pipeline will run using the <code>andersenlab/gatk4</code> docker image. You must have docker installed.</p>
<pre><code>nextflow run andersenlab/wi-gatk -profile local -resume
</code></pre>
<h2 id="debugging_the_pipeline_on_quest">Debugging the pipeline on Quest<a class="headerlink" href="#debugging_the_pipeline_on_quest" title="Permanent link">&para;</a></h2>
<p>When running on Quest, you should first run the quest debug profile. The Quest debug profile will use a test dataset and sample sheet which runs much faster and will encounter errors much sooner should they need to be fixed. If the debug dataset runs to completion it is likely that the full dataset will as well.</p>
<pre><code>nextflow run andersenlab/wi-gatk -profile debug -resume
</code></pre>
<h2 id="running_the_pipeline_on_quest">Running the pipeline on Quest<a class="headerlink" href="#running_the_pipeline_on_quest" title="Permanent link">&para;</a></h2>
<p>The pipeline can be run on Quest using the following command:</p>
<pre><code>nextflow run andersenlab/wi-gatk -profile quest --sample_sheet &lt;path_to_sheet&gt;
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<p>Most configuration is handled using the <code>-profile</code> flag and <code>nextflow.config</code>; If you want to fine tune things you can use the options below.</p>
<h2 id="--sample_sheet">--sample_sheet<a class="headerlink" href="#--sample_sheet" title="Permanent link">&para;</a></h2>
<p>The sample sheet is automatically generated from <code>alignment-nf</code>. The sample sheet contains 5 columns as detailed below:</p>
<table>
<thead>
<tr>
<th style="text-align: left;">strain</th>
<th style="text-align: left;">bam</th>
<th style="text-align: left;">bai</th>
<th style="text-align: left;">coverage</th>
<th style="text-align: left;">percent_mapped</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">AB1</td>
<td style="text-align: left;">AB1.bam</td>
<td style="text-align: left;">AB1.bam.bai</td>
<td style="text-align: left;">64</td>
<td style="text-align: left;">99.4</td>
</tr>
<tr>
<td style="text-align: left;">AB4</td>
<td style="text-align: left;">AB4.bam</td>
<td style="text-align: left;">AB4.bam.bai</td>
<td style="text-align: left;">52</td>
<td style="text-align: left;">99.2</td>
</tr>
<tr>
<td style="text-align: left;">BRC20067</td>
<td style="text-align: left;">BRC20067.bam</td>
<td style="text-align: left;">BRC20067.bam.bai</td>
<td style="text-align: left;">30</td>
<td style="text-align: left;">92.5</td>
</tr>
</tbody>
</table>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>It is essential that you always use the pipelines and scripts to generate this sample sheet and <strong>NEVER</strong> manually. There are lots of strains and we want to make sure the entire process can be reproduced.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The sample sheet produced from <code>alignment-nf</code> is only for strains that you ran in the alignment pipeline most recently. If you want to combine old strains with new strains, you will have to combine two or more sample sheets. <strong>If you are running a species-wide analysis for CeNDR, please follow the notes in the full WI protocol <a href="https://katiesevans9.notion.site/Wild-isolate-sequence-analysis-protocol-00e76cc7f55f4bf6ab644dd99c883727">here</a></strong></p>
</div>
<h3 id="--bam_location_optional">--bam_location (optional)<a class="headerlink" href="#--bam_location_optional" title="Permanent link">&para;</a></h3>
<p>Path to directory holding all the alignment files for strains in the analysis. Defaults to <code>/projects/b1059/data/{species}/WI/alignments/</code></p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Remember to move your bam files output from <a href="github.com/andersenlab/alignment-nf">alignment-nf</a> to this location prior to running <code>wi-gatk</code>. In <strong>most</strong> cases, you will want to run <code>wi-gatk</code> on all samples, new and old combined.</p>
</div>
<h3 id="--species_optional">--species (optional)<a class="headerlink" href="#--species_optional" title="Permanent link">&para;</a></h3>
<p><strong>default</strong> = c_elegans</p>
<p>Options: c_elegans, c_briggsae, or c_tropicalis</p>
<h3 id="--project_optional">--project (optional)<a class="headerlink" href="#--project_optional" title="Permanent link">&para;</a></h3>
<p><strong>default</strong> = PRJNA13758</p>
<p>WormBase project ID for selected species. Choose from some examples <a href="https://github.com/AndersenLab/genomes-nf/blob/master/bin/project_species.tsv">here</a></p>
<h3 id="--ws_build_optional">--ws_build (optional)<a class="headerlink" href="#--ws_build_optional" title="Permanent link">&para;</a></h3>
<p><strong>default</strong> = WS283</p>
<p>WormBase version to use for reference genome.</p>
<h3 id="--reference_optional">--reference (optional)<a class="headerlink" href="#--reference_optional" title="Permanent link">&para;</a></h3>
<p>A fasta reference indexed with BWA. On Quest, the reference is available here:</p>
<pre><code>/projects/b1059/data/c_elegans/genomes/PRJNA13758/WS283/c_elegans.PRJNA13758.WS283.genome.fa.gz
</code></pre>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If running on QUEST, instead of changing the <code>reference</code> parameter, opt to change the <code>species</code>, <code>project</code>, and <code>ws_build</code> for other species like c_briggsae (and then the reference will change automatically) </p>
</div>
<h3 id="--mito_name_optional">--mito_name (optional)<a class="headerlink" href="#--mito_name_optional" title="Permanent link">&para;</a></h3>
<p>Name of contig to skip het polarization. Might need to change for other species besides c_elegans if the mitochondria contig is named differently. Defaults to <code>MtDNA</code>.</p>
<h3 id="--output_optional">--output (optional)<a class="headerlink" href="#--output_optional" title="Permanent link">&para;</a></h3>
<p>A directory in which to output results. By default it will be <code>WI-YYYYMMDD</code> where YYYYMMDD is todays date.</p>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<p>The final output directory looks like this:</p>
<pre><code>
├── variation
│   ├── *.hard-filter.vcf.gz
│   ├── *.hard-filter.vcf.tbi
│   ├── *.hard-filter.stats.txt
│   ├── *.hard-filter.filter_stats.txt
│   ├── *.soft-filter.vcf.gz
│   ├── *.soft-filter.vcf.tbi
│   ├── *.soft-filter.stats.txt
│   └── *.soft-filter.filter_stats.txt
└── report
    ├── multiqc.html
    └── multiqc_data
        └── multiqc_*.json
   
</code></pre>
<h1 id="data_storage">Data Storage<a class="headerlink" href="#data_storage" title="Permanent link">&para;</a></h1>
<h2 id="cleanup">Cleanup<a class="headerlink" href="#cleanup" title="Permanent link">&para;</a></h2>
<p>The <code>hard-filter.vcf</code> is the input for both the <a href="https://github.com/AndersenLab/concordance-nf"><code>concordance-nf</code></a> pipeline and the <a href="https://github.com/AndersenLab/post-gatk-nf"><code>post-gatk-nf</code></a> pipeline. Once both pipelines have been completed successfully, the hard and soft filter vcf and index files (everything output in the <code>variation</code> folder) can be moved to <code>/projects/b1059/data/{species}/WI/variation/{date}/vcf</code>.</p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../pipeline-alignment/" class="btn btn-neutral float-left" title="alignment-nf"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../pipeline-concordance/" class="btn btn-neutral float-right" title="concordance-nf">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-alignment/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../pipeline-concordance/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="../search/main.js"></script>
      <script src="../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
