<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>genomes-nf - Andersen Lab Dry Guide</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "genomes-nf";
        var mkdocs_page_input_path = "pipelines/pipeline-genomes-nf.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Andersen Lab Dry Guide
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Home</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../..">Overview</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Knowledge Base</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/best_practices/">Best Practices</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/r/">R</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/shiny/">R Shiny</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/bash/">Command Line</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/github/">Git and Github</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Rockfish</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/VAST/">Data on Rockfish</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quest</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/b1059/">Data on b1059</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Nextflow Pipelines</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" >WI Sequencing pipelines</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" >Other pipelines</a>
    <ul class="current">
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                </li>
                <li class="toctree-l2 current"><a class="reference internal current" href="#">genomes-nf</a>
    <ul class="current">
    </ul>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../nf-GCPconfig/">GCP</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other things</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/caendr/">CaeNDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/backup/">Backup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sra/">SRA</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/cloud/">Cloud</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sharing-globus/">Globus file sharing</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/writing-nextflow/">Writing Nextflow workflows</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/labsite/">Andersen Labsite</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Andersen Lab Dry Guide</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Nextflow Pipelines</li>
          <li class="breadcrumb-item">Other pipelines</li>
      <li class="breadcrumb-item active">genomes-nf</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipelines/pipeline-genomes-nf.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="genomes-nf">genomes-nf<a class="headerlink" href="#genomes-nf" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#genomes-nf">genomes-nf</a></li>
<li><a href="#overview">Overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a><ul>
<li><a href="#relevant_docker_images">Relevant Docker Images</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#testing_on_rockfish">Testing on Rockfish</a></li>
<li><a href="#running_on_rockfish">Running on Rockfish</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#-profile">-profile</a></li>
<li><a href="#-wb_version">-wb_version</a></li>
<li><a href="#--projects">--projects</a></li>
</ul>
</li>
<li><a href="#output">Output</a><ul>
<li><a href="#notes">Notes</a></li>
</ul>
</li>
</ul>
</div>
<h1 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h1>
<p>This repo contains a nextflow pipeline that downloads, indexes, and builds annotation databases for reference genomes from wormbase. The following outputs are created:</p>
<ol>
<li>A BWA Index</li>
<li>SNPeff annotation database</li>
<li>CSQ annotation database</li>
<li>Samtools faidx index</li>
<li>A GATK Sequence dictionary file</li>
</ol>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 23+. On Rockfish, you can access this version by loading the <code>nf23_env</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda
source activate /data/eande106/software/conda_envs/nf23_env
</code></pre>
<h3 id="relevant_docker_images">Relevant Docker Images<a class="headerlink" href="#relevant_docker_images" title="Permanent link">&para;</a></h3>
<ul>
<li><code>andersenlab/genomes</code> (<a href="https://hub.docker.com/r/andersenlab/genomes-nf">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/genomes.Dockerfile</code> or <code>.github/workflows/build.yml</code> GitHub actions will create a new docker image and push if successful.</li>
</ul>
<p>Make sure that you add the following code to your <code>~/.bash_profile</code>. This line makes sure that any singularity images you download will go to a shared location on <code>/vast/eande106</code> for other users to take advantage of (without them also having to download the same image).</p>
<pre><code># add singularity cache
export SINGULARITY_CACHEDIR='/vast/eande106/singularity/'
</code></pre>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you need to work with the docker container, you will need to create an interactive session as singularity can't be run on Rockfish login nodes.</p>
<p><code>interact -n1 -pexpress
module load singularity
singularity shell [--bind local_dir:container_dir] /vast/eande106/singularity/&lt;image_name&gt;</code></p>
</div>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../../rockfish/rf-nextflow/">Nextflow</a> page.</em></p>
<h2 id="testing_on_rockfish">Testing on Rockfish<a class="headerlink" href="#testing_on_rockfish" title="Permanent link">&para;</a></h2>
<p><em>This command uses a test dataset</em></p>
<pre><code>nextflow run -latest andersenlab/genomes-nf --debug
</code></pre>
<h2 id="running_on_rockfish">Running on Rockfish<a class="headerlink" href="#running_on_rockfish" title="Permanent link">&para;</a></h2>
<p>You should run this in a screen or tmux session.</p>
<pre><code>nextflow run -latest andersenlab/genomes-nf -resume --wb_version=WS276 --projects=c_elegans/PRJNA13758
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<h2 id="-profile"><code>-profile</code><a class="headerlink" href="#-profile" title="Permanent link">&para;</a></h2>
<p>Can be set to 'rockfish' (default), <code>local</code>, or <code>quest</code>.</p>
<h2 id="-wb_version"><code>-wb_version</code><a class="headerlink" href="#-wb_version" title="Permanent link">&para;</a></h2>
<p>The wormbase version to build. For example, <code>WS276</code>.</p>
<h2 id="--projects"><code>--projects</code><a class="headerlink" href="#--projects" title="Permanent link">&para;</a></h2>
<p>A comma-delimited list of <code>species/project_id</code> identifiers. A table below lists the current projects that can be downloaded. This table is regenerated as the first step of the pipeline, and stored as a file called <code>project_species.tsv</code> in the <code>params.output</code> folder (<code>./genomes</code> if working locally).</p>
<p>The current set of available species/projects that can be built are:</p>
<table>
<thead>
<tr>
<th style="text-align: left;">species</th>
<th style="text-align: left;">project</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">b_xylophilus</td>
<td style="text-align: left;">PRJEA64437</td>
</tr>
<tr>
<td style="text-align: left;">c_briggsae</td>
<td style="text-align: left;">PRJNA10731</td>
</tr>
<tr>
<td style="text-align: left;">c_angaria</td>
<td style="text-align: left;">PRJNA51225</td>
</tr>
<tr>
<td style="text-align: left;">a_ceylanicum</td>
<td style="text-align: left;">PRJNA231479</td>
</tr>
<tr>
<td style="text-align: left;">a_suum</td>
<td style="text-align: left;">PRJNA62057</td>
</tr>
<tr>
<td style="text-align: left;">a_suum</td>
<td style="text-align: left;">PRJNA80881</td>
</tr>
<tr>
<td style="text-align: left;">b_malayi</td>
<td style="text-align: left;">PRJNA10729</td>
</tr>
<tr>
<td style="text-align: left;">c_brenneri</td>
<td style="text-align: left;">PRJNA20035</td>
</tr>
<tr>
<td style="text-align: left;">c_elegans</td>
<td style="text-align: left;">PRJEB28388</td>
</tr>
<tr>
<td style="text-align: left;">c_elegans</td>
<td style="text-align: left;">PRJNA13758</td>
</tr>
<tr>
<td style="text-align: left;">c_elegans</td>
<td style="text-align: left;">PRJNA275000</td>
</tr>
<tr>
<td style="text-align: left;">c_latens</td>
<td style="text-align: left;">PRJNA248912</td>
</tr>
<tr>
<td style="text-align: left;">c_remanei</td>
<td style="text-align: left;">PRJNA248909</td>
</tr>
<tr>
<td style="text-align: left;">c_remanei</td>
<td style="text-align: left;">PRJNA248911</td>
</tr>
<tr>
<td style="text-align: left;">c_remanei</td>
<td style="text-align: left;">PRJNA53967</td>
</tr>
<tr>
<td style="text-align: left;">c_inopinata</td>
<td style="text-align: left;">PRJDB5687</td>
</tr>
<tr>
<td style="text-align: left;">c_japonica</td>
<td style="text-align: left;">PRJNA12591</td>
</tr>
<tr>
<td style="text-align: left;">c_sp11</td>
<td style="text-align: left;">PRJNA53597</td>
</tr>
<tr>
<td style="text-align: left;">c_sp5</td>
<td style="text-align: left;">PRJNA194557</td>
</tr>
<tr>
<td style="text-align: left;">c_nigoni</td>
<td style="text-align: left;">PRJNA384657</td>
</tr>
<tr>
<td style="text-align: left;">c_sinica</td>
<td style="text-align: left;">PRJNA194557</td>
</tr>
<tr>
<td style="text-align: left;">c_tropicalis</td>
<td style="text-align: left;">PRJNA53597</td>
</tr>
<tr>
<td style="text-align: left;">d_immitis</td>
<td style="text-align: left;">PRJEB1797</td>
</tr>
<tr>
<td style="text-align: left;">h_bacteriophora</td>
<td style="text-align: left;">PRJNA13977</td>
</tr>
<tr>
<td style="text-align: left;">l_loa</td>
<td style="text-align: left;">PRJNA60051</td>
</tr>
<tr>
<td style="text-align: left;">m_hapla</td>
<td style="text-align: left;">PRJNA29083</td>
</tr>
<tr>
<td style="text-align: left;">m_incognita</td>
<td style="text-align: left;">PRJEA28837</td>
</tr>
<tr>
<td style="text-align: left;">h_contortus</td>
<td style="text-align: left;">PRJEB506</td>
</tr>
<tr>
<td style="text-align: left;">h_contortus</td>
<td style="text-align: left;">PRJNA205202</td>
</tr>
<tr>
<td style="text-align: left;">n_americanus</td>
<td style="text-align: left;">PRJNA72135</td>
</tr>
<tr>
<td style="text-align: left;">p_exspectatus</td>
<td style="text-align: left;">PRJEB6009</td>
</tr>
<tr>
<td style="text-align: left;">o_tipulae</td>
<td style="text-align: left;">PRJEB15512</td>
</tr>
<tr>
<td style="text-align: left;">p_redivivus</td>
<td style="text-align: left;">PRJNA186477</td>
</tr>
<tr>
<td style="text-align: left;">s_ratti</td>
<td style="text-align: left;">PRJEA62033</td>
</tr>
<tr>
<td style="text-align: left;">s_ratti</td>
<td style="text-align: left;">PRJEB125</td>
</tr>
<tr>
<td style="text-align: left;">o_volvulus</td>
<td style="text-align: left;">PRJEB513</td>
</tr>
<tr>
<td style="text-align: left;">p_pacificus</td>
<td style="text-align: left;">PRJNA12644</td>
</tr>
<tr>
<td style="text-align: left;">t_muris</td>
<td style="text-align: left;">PRJEB126</td>
</tr>
<tr>
<td style="text-align: left;">t_spiralis</td>
<td style="text-align: left;">PRJNA12603</td>
</tr>
<tr>
<td style="text-align: left;">t_suis</td>
<td style="text-align: left;">PRJNA208415</td>
</tr>
<tr>
<td style="text-align: left;">t_suis</td>
<td style="text-align: left;">PRJNA208416</td>
</tr>
</tbody>
</table>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<p>Outputs are nested under <code>params.output</code> with the following structure:</p>
<pre><code>c_elegans                                                                   (species)
└── genomes
    └── PRJNA13758                                                          (project)
        └── WS276                                                           (build)
            ├── c_elegans.PRJNA13758.WS276.genome.dict                      (dict file)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz                     (fasta)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.amb                 (bwa index)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.ann                 (bwa index)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.bwt                 (bwa index)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.fai                 (samtools faidx index)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.gzi                 (bwa index)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.pac                 (bwa index)
            ├── c_elegans.PRJNA13758.WS276.genome.fa.gz.sa                  (bwa index)
            ├── csq
            │   ├── c_elegans.PRJNA13758.WS276.csq.gff3.gz                  (CSQ annotation GFF3)
            │   ├── c_elegans.PRJNA13758.WS276.csq.gff3.gz.tbi              (tabix index)
            │   ├── c_elegans.PRJNA13758.WS276.AA_Length.tsv                (protein lengths)
            │   └── c_elegans.PRJNA13758.WS276.AA_Scores.tsv                (blosum and grantham scores)
            ├── lcr
            │   ├── c_elegans.PRJNA13758.WS276.repeat_masker.bed.gz         (low complexity regions)
            │   ├── c_elegans.PRJNA13758.WS276.repeat_masker.bed.gz.tbi     (tabix index)
            │   ├── c_elegans.PRJNA13758.WS276.dust.bed.gz                  (low complexity regions)
            │   └── c_elegans.PRJNA13758.WS276.dust.bed.gz.tbi              (tabix index)
            └── snpeff
                ├── c_elegans.PRJNA13758.WS276                              (tabix index)
                │   ├── genes.gtf.gz                                        (Reference GTF)
                │   ├── sequences.fa                                        (fasta genome (unzipped))
                │   └── snpEffectPredictor.bin                              (snpEff annotation db)
                └── snpEff.config                                           (snpEff configuration file)

</code></pre>
<h2 id="notes">Notes<a class="headerlink" href="#notes" title="Permanent link">&para;</a></h2>
<ul>
<li>The SNPeff databases are not collected together in one location as is often the case. Instead, they are stored individually with their own configuration files.</li>
<li>The GFF3 files for some species are not as developed as <em>C. elegans</em>. As a consequence, the biotype is inferred from the Attributes column of the GFF. See <code>bin/format_csq.R</code> for more details.</li>
</ul>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The updated csq-formated gff script needs to be updated for other species besides <em>C. elegans</em> (if running the default mode)</p>
</div>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../pipeline-nil-ril/" class="btn btn-neutral float-left" title="nil-ril-nf"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../docker/" class="btn btn-neutral float-right" title="Docker">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-nil-ril/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../docker/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/main.js"></script>
      <script src="../../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
