<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>annotation-nf - Andersen Lab Dry Guide</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  <link href="../css/version-select.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "annotation-nf";
    var mkdocs_page_input_path = "pipeline-annotation-nf.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Andersen Lab Dry Guide</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Knowledge Base</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../best_practices/">Best Practices</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../r/">R</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../shiny/">R Shiny</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../bash/">Command Line</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../github/">Git and Github</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Quest</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-intro/">Introduction</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../b1059/">Data on b1059</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-nextflow/">Nextflow</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-conda/">Conda</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-docker/">Docker</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-GCPconfig/">GCP</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">WI Sequencing pipelines</span></p>
                <ul class="current">
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                    </li>
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">annotation-nf</a>
    <ul class="current">
    </ul>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Other pipelines</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-cegwas/">cegwas2-nf</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Other things</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../cendr/">CeNDR</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../backup/">Backup</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../cloud/">Cloud</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../sharing-globus/">Globus file sharing</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../labsite/">Andersen Labsite</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Andersen Lab Dry Guide</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>WI Sequencing pipelines &raquo;</li>
        
      
    
    <li>annotation-nf</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipeline-annotation-nf.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h1 id="annotation-nf">annotation-nf<a class="headerlink" href="#annotation-nf" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#annotation-nf">annotation-nf</a></li>
<li><a href="#pipeline_overview">Pipeline overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a></li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#testing_on_quest">Testing on Quest</a></li>
<li><a href="#running_on_quest">Running on Quest</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#--debug">--debug</a></li>
<li><a href="#--vcf">--vcf</a></li>
<li><a href="#--species">--species</a></li>
<li><a href="#--divergent_regions">--divergent_regions</a><ul>
<li><a href="#--reference_--project_--ws_build_optional">--reference, --project, --ws_build (optional)</a></li>
<li><a href="#--ncsq_param_optional">--ncsq_param (optional)</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#output">Output</a></li>
<li><a href="#data_storage">Data storage</a><ul>
<li><a href="#cleanup">Cleanup</a></li>
<li><a href="#updating_cendr">Updating CeNDR</a></li>
<li><a href="#updating_nemascan">Updating NemaScan</a></li>
</ul>
</li>
</ul>
</div>
<p>The <a href="https://github.com/AndersenLab/annotation-nf">annotation-nf</a> pipeline performs variant annotation for the VCF at the isotype level using both <a href="http://pcingola.github.io/SnpEff/">SnpEff</a> and <a href="https://samtools.github.io/bcftools/howtos/csq-calling.html">BCFtools/csq (BCSQ)</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Before running, make sure to check out the <a href="../pipeline-genomes-nf/">genomes-nf</a> pipeline to ensure that the reference genome and annotation databases are set up properly.</p>
</div>
<h1 id="pipeline_overview">Pipeline overview<a class="headerlink" href="#pipeline_overview" title="Permanent link">&para;</a></h1>
<pre><code>-------------    
ANNOTATION-NF
-------------

nextflow main.nf --debug

nextflow main.nf --vcf=hard-filtered.vcf --species=c_elegans --divergent_regions=divergent_regions_strain.bed

    parameters           description                                              Set/Default
    ==========           ===========                                              ========================
    --debug              Set to 'true' to test                                    ${params.debug}
    --species            Species: 'c_elegans', 'c_tropicalis' or 'c_briggsae'     ${params.species}
    --vcf                hard filtered vcf to calculate variant density           ${params.vcf}
    --divergent_regions  (Optional) Divergent region bed file                     ${params.divergent_regions}
    --reference          Reference used based on species and project              ${params.reference}
    --output             (Optional) output folder name                            ${params.output}

    username                                                                      ${&quot;whoami&quot;.execute().in.text}

    HELP: http://andersenlab.org/dry-guide/pipeline-annotation-nf
</code></pre>
<p><img alt="Pipeline-overview" src="../img/annotation-nf.drawio.svg" /></p>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 20.0+. On QUEST, you can access this version by loading the <code>nf20_env</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda3.6
source activate /projects/b1059/software/conda_envs/nf20_env
</code></pre>
<p>Alternatively you can update Nextflow by running:</p>
<pre><code>nextflow self-update
</code></pre>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>This pipeline currently only supports analysis on Quest, cannot be run locally</p>
</div>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
</div>
<p>For more info about running Nextflow pipelines in the Andersen Lab, check out <a href="../quest-nextflow/">this page</a></p>
<h2 id="testing_on_quest">Testing on Quest<a class="headerlink" href="#testing_on_quest" title="Permanent link">&para;</a></h2>
<p><em>This command uses a test dataset</em></p>
<pre><code>nextflow run andersenlab/annotation-nf --debug
</code></pre>
<h2 id="running_on_quest">Running on Quest<a class="headerlink" href="#running_on_quest" title="Permanent link">&para;</a></h2>
<p>You should run this in a screen session.</p>
<pre><code>nextflow run andersenlab/annotation-nf --vcf &lt;path_to_vcf&gt; --species &lt;species&gt; --divergent_regions &lt;path_to_file&gt;
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<h2 id="--debug">--debug<a class="headerlink" href="#--debug" title="Permanent link">&para;</a></h2>
<p>You should use <code>--debug true</code> for testing/debugging purposes. This will run the debug test set (located in the <code>test_data</code> folder).</p>
<p>For example:</p>
<pre><code>nextflow run andersenlab/annotation-nf --debug
</code></pre>
<h2 id="--vcf">--vcf<a class="headerlink" href="#--vcf" title="Permanent link">&para;</a></h2>
<p>Path to the hard-filter, isotype VCF (output from <code>post-gatk-nf</code>)</p>
<h2 id="--species">--species<a class="headerlink" href="#--species" title="Permanent link">&para;</a></h2>
<p>Choose from c_elegans, c_briggsae, or c_tropicalis. Species will specifiy a default reference genome. You can select a different one if you prefer (see below)</p>
<h2 id="--divergent_regions">--divergent_regions<a class="headerlink" href="#--divergent_regions" title="Permanent link">&para;</a></h2>
<p>This is the <code>divergent_regions_strain.bed</code> file output from the <code>post-gatk-nf</code> pipeline. This file is used to add a column to the flat file if the variant is within a divergent region. Currently, <em>C. elegans</em> is the only species with divergent regions, if running for another species, do not provide a divergent_regions file and the pipeline will ignore it.</p>
<h3 id="--reference_--project_--ws_build_optional">--reference, --project, --ws_build (optional)<a class="headerlink" href="#--reference_--project_--ws_build_optional" title="Permanent link">&para;</a></h3>
<p>By default, the reference genome is set by the species parameter. If you don't want to use the default, you could change the project and/or ws_build. As long as the genome is in the proper location on quest (for more, see the <a href="../pipeline-genomes-nf/">genomes-nf</a> pipeline), this will work. Alternatively, you could provide the path to a reference of your choice.</p>
<p><strong>Defaults:</strong>
- <em>C. elegans</em> - <code>/projects/b1059/data/c_elegans/genomes/PRJNA13758/WS276/c_elegans.PRJNA13758.WS276.genome.fa.gz</code>
- <em>C. briggsae</em> - <code>/projects/b1059/data/c_briggsae/genomes/QX1410_nanopore/Feb2020/c_briggsae.QX1410_nanopore.Feb2020.genome.fa.gz</code>
- <em>C. tropicalis</em> - <code>/projects/b1059/data/c_tropicalis/genomes/NIC58_nanopore/June2021/c_tropicalis.NIC58_nanopore.June2021.genome.fa.gz</code></p>
<h3 id="--ncsq_param_optional">--ncsq_param (optional)<a class="headerlink" href="#--ncsq_param_optional" title="Permanent link">&para;</a></h3>
<p>This parameter is necessary for correct annotation using BCSQ for variants with many different annotations (like found in divergent regions). In 20210121 we found that the default value of <code>224</code> was sufficient, but as more strains are added this number might need to increase. If there is an issue, you should see a warning error from BCFtools and they should suggest what to change this parameter to.</p>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<pre><code>├── strain_vcf
│   ├── {strain}.{date}.vcf.gz
│   └── {strain}.{date}.vcf.gz.tbi
└── variation
    ├── WI.{date}.hard-filter.isotype.snpeff.vcf.gz
    ├── WI.{date}.hard-filter.isotype.snpeff.vcf.gz.tbi
    ├── snpeff.stats.csv
    ├── WI.{date}.hard-filter.isotype.bcsq.vcf.gz
    ├── WI.{date}.hard-filter.isotype.bcsq.vcf.gz.tbi
    ├── WI.{date}.hard-filter.isotype.bcsq.vcf.gz.stats.txt 
    └── WI.{date}.strain-annotation.bcsq.tsv

</code></pre>
<h1 id="data_storage">Data storage<a class="headerlink" href="#data_storage" title="Permanent link">&para;</a></h1>
<h2 id="cleanup">Cleanup<a class="headerlink" href="#cleanup" title="Permanent link">&para;</a></h2>
<p>Once the pipeline has complete successfully and you are satisfied with the results, the final data can be moved to their final storage place on Quest accordingly:</p>
<ul>
<li>Both the <code>strain_vcf</code> and the <code>variation</code> folders can be moved to <code>/projects/b1059/data/{species}/WI/variation/{date}/vcf</code></li>
<li>If applicable, all snpeff <code>.bed</code> files (HIGH, LOW, MODERATE, etc.) can be moved to <code>/projects/b1059/data/{species}/WI/variation/{date}/tracks/</code> (<em>As of 20210901 this is no longer being produced for CeNDR</em>)</li>
</ul>
<h2 id="updating_cendr">Updating CeNDR<a class="headerlink" href="#updating_cendr" title="Permanent link">&para;</a></h2>
<p>Check out the <a href="../cendr/">CeNDR</a> page for more information about updating a new data release for CeNDR.</p>
<h2 id="updating_nemascan">Updating <code>NemaScan</code><a class="headerlink" href="#updating_nemascan" title="Permanent link">&para;</a></h2>
<p>Once a new CeNDR release is ready, it is important to also update the genome-wide association mapping packages to ensure users can appropriately analyze data from new strains as well as old strains. Here is a list of things that need to be updated:</p>
<ul>
<li>The default vcf should be changed to the newest release date (i.e. from 20200815 to 20210121). Users will still have the option to use an earlier vcf.</li>
<li>Ensure strain annotation flat file is stored in the proper file location to be accessed, both on QUEST and on GCP for CeNDR/local users.</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../pipeline-nemascan/" class="btn btn-neutral float-right" title="NemaScan">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../pipeline-impute/" class="btn btn-neutral" title="imputation"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide/" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-impute/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../pipeline-nemascan/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
      <script src="../js/version-select.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
