<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>wi-gatk - Andersen Lab Dry Guide</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "wi-gatk";
        var mkdocs_page_input_path = "pipelines/pipeline-wiGATK.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Andersen Lab Dry Guide
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Home</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../..">Overview</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Knowledge Base</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/best_practices/">Best Practices</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/r/">R</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/shiny/">R Shiny</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/bash/">Command Line</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/github/">Git and Github</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Rockfish</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/VAST/">Data on Rockfish</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quest</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/b1059/">Data on b1059</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Nextflow Pipelines</span></p>
              <ul class="current">
                  <li class="toctree-l1 current"><a class="reference internal current" >WI Sequencing pipelines</a>
    <ul class="current">
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                </li>
                <li class="toctree-l2 current"><a class="reference internal current" href="#">wi-gatk</a>
    <ul class="current">
    </ul>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" >Other pipelines</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../nf-GCPconfig/">GCP</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other things</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/caendr/">CaeNDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/backup/">Backup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sra/">SRA</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/cloud/">Cloud</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sharing-globus/">Globus file sharing</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/writing-nextflow/">Writing Nextflow workflows</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/labsite/">Andersen Labsite</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Andersen Lab Dry Guide</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Nextflow Pipelines</li>
          <li class="breadcrumb-item">WI Sequencing pipelines</li>
      <li class="breadcrumb-item active">wi-gatk</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipelines/pipeline-wiGATK.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="wi-gatk">wi-gatk<a class="headerlink" href="#wi-gatk" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#wi-gatk">wi-gatk</a></li>
<li><a href="#pipeline_overview">Pipeline Overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a></li>
<li><a href="#relevant_docker_images">Relevant Docker Images</a></li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#testing_on_rockfish">Testing on Rockfish</a></li>
<li><a href="#running_on_rockfish">Running on Rockfish</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#-profile">-profile</a></li>
<li><a href="#--sample_sheet">--sample_sheet</a></li>
<li><a href="#--bam_location_optional">--bam_location (optional)</a></li>
<li><a href="#--species_optional">--species (optional)</a></li>
<li><a href="#--project_optional">--project (optional)</a></li>
<li><a href="#--ws_build_optional">--ws_build (optional)</a></li>
<li><a href="#--reference_optional">--reference (optional)</a></li>
<li><a href="#--mito_name_optional">--mito_name (optional)</a></li>
<li><a href="#--output_optional">--output (optional)</a></li>
</ul>
</li>
<li><a href="#output">Output</a></li>
<li><a href="#data_storage">Data Storage</a><ul>
<li><a href="#cleanup">Cleanup</a></li>
</ul>
</li>
</ul>
</div>
<p>The <code>wi-gatk</code> pipeline filters and calls variants from wild isolate sequence data.</p>
<h1 id="pipeline_overview">Pipeline Overview<a class="headerlink" href="#pipeline_overview" title="Permanent link">&para;</a></h1>
<pre><code>
    _______ _______ _______ __  __      _______ _______ 
    |     __|   _   |_     _|  |/  |    |    |  |    ___|
    |    |  |       | |   | |     &lt;     |       |    ___|
    |_______|___|___| |___| |__|\__|    |__|____|___|    


    parameters                 description                           Set/Default
    ==========                 ===========                           ========================
    --debug                    Use --debug to indicate debug mode    null
    --output                   Release Directory                     WI-{date}
    --sample_sheet             Sample sheet                          null
    --bam_location             Directory of bam files                /projects/b1059/data/{species}/WI/alignments/
    --mito_name                Contig not to polarize hetero sites   MtDNA

    Reference Genome
    --------------- 
    --reference_base           Location of ref genomes               /projects/b1059/data/{species}/genomes/
    --species/project/build    These 4 params form --reference       {species} / {project} / {ws_build}

    Variant Filters         
    ---------------           
    --min_depth                Minimum variant depth                 5
    --qual                     Variant QUAL score                    30
    --strand_odds_ratio        SOR_strand_odds_ratio                 5
    --quality_by_depth         QD_quality_by_depth                   20
    --fisherstrand             FS_fisher_strand                      100
    --high_missing             Max % missing genotypes               0.95
    --high_heterozygosity      Max % max heterozygosity              0.10


</code></pre>
<p><img alt="Pipeline-overview" src="../../img/wi-gatk.drawio.svg" /></p>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 23+. On Rockfish, you can access this version by loading the <code>nf23_env</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda
source activate /data/eande106/software/conda_envs/nf23_env
</code></pre>
<h2 id="relevant_docker_images">Relevant Docker Images<a class="headerlink" href="#relevant_docker_images" title="Permanent link">&para;</a></h2>
<ul>
<li><code>andersenlab/gatk4</code> (<a href="https://hub.docker.com/r/andersenlab/gatk4">link</a>): Docker image is created within this pipeline using GitHub actions. Whenever a change is made to <code>env/gatk4.Dockerfile</code> or <code>.github/workflows/build_docker.yml</code> GitHub actions will create a new docker image and push if successful</li>
<li><code>andersenlab/r_packages</code> (<a href="https://hub.docker.com/r/andersenlab/r_packages">link</a>): Docker image is created manually, code can be found in the <a href="https://github.com/AndersenLab/dockerfile/tree/master/r_packages">dockerfile</a> repo.</li>
</ul>
<p>Make sure that you add the following code to your <code>~/.bash_profile</code>. This line makes sure that any singularity images you download will go to a shared location on <code>/vast/eande106</code> for other users to take advantage of (without them also having to download the same image).</p>
<pre><code># add singularity cache
export SINGULARITY_CACHEDIR='/vast/eande106/singularity/'
</code></pre>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you need to work with the docker container, you will need to create an interactive session as singularity can't be run on Rockfish login nodes.</p>
<p><code>interact -n1 -pexpress
module load singularity
singularity shell [--bind local_dir:container_dir] /vast/eande106/singularity/&lt;image_name&gt;</code></p>
</div>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../../rockfish/rf-nextflow/">Nextflow</a> page.</em></p>
<h2 id="testing_on_rockfish">Testing on Rockfish<a class="headerlink" href="#testing_on_rockfish" title="Permanent link">&para;</a></h2>
<p><em>This command uses a test dataset</em></p>
<pre><code>nextflow run -latest andersenlab/wi-gatk --debug
</code></pre>
<h2 id="running_on_rockfish">Running on Rockfish<a class="headerlink" href="#running_on_rockfish" title="Permanent link">&para;</a></h2>
<p>You should run this in a screen or tmux session.</p>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../../rockfish/rf-nextflow/">Nextflow</a> page.</em></p>
<pre><code>nextflow run -latest andersenlab/wi-gatk --sample_sheet &lt;path_to_sample_sheet&gt; --species c_elegans
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<h2 id="-profile">-profile<a class="headerlink" href="#-profile" title="Permanent link">&para;</a></h2>
<p>There are three configuration profiles for this pipeline.</p>
<ul>
<li><code>rockfish</code> - Used for running on Rockfish (default).</li>
<li><code>quest</code>    - Used for running on Quest.</li>
<li><code>local</code>    - Used for local development.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you forget to add a <code>-profile</code>, the <code>rockfish</code> profile will be chosen as default</p>
</div>
<h2 id="--sample_sheet">--sample_sheet<a class="headerlink" href="#--sample_sheet" title="Permanent link">&para;</a></h2>
<p>The sample sheet is automatically generated from <code>alignment-nf</code>. The sample sheet contains 5 columns as detailed below:</p>
<table>
<thead>
<tr>
<th style="text-align: left;">strain</th>
<th style="text-align: left;">bam</th>
<th style="text-align: left;">bai</th>
<th style="text-align: left;">coverage</th>
<th style="text-align: left;">percent_mapped</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">AB1</td>
<td style="text-align: left;">AB1.bam</td>
<td style="text-align: left;">AB1.bam.bai</td>
<td style="text-align: left;">64</td>
<td style="text-align: left;">99.4</td>
</tr>
<tr>
<td style="text-align: left;">AB4</td>
<td style="text-align: left;">AB4.bam</td>
<td style="text-align: left;">AB4.bam.bai</td>
<td style="text-align: left;">52</td>
<td style="text-align: left;">99.2</td>
</tr>
<tr>
<td style="text-align: left;">BRC20067</td>
<td style="text-align: left;">BRC20067.bam</td>
<td style="text-align: left;">BRC20067.bam.bai</td>
<td style="text-align: left;">30</td>
<td style="text-align: left;">92.5</td>
</tr>
</tbody>
</table>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>It is essential that you always use the pipelines and scripts to generate this sample sheet and <strong>NEVER</strong> manually. There are lots of strains and we want to make sure the entire process can be reproduced.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The sample sheet produced from <code>alignment-nf</code> is only for strains that you ran in the alignment pipeline most recently. If you want to combine old strains with new strains, you will have to combine two or more sample sheets. <strong>If you are running a species-wide analysis for CaeNDR, please follow the notes in the full WI protocol <a href="https://katiesevans9.notion.site/Wild-isolate-sequence-analysis-protocol-00e76cc7f55f4bf6ab644dd99c883727">here</a></strong></p>
</div>
<h2 id="--bam_location_optional">--bam_location (optional)<a class="headerlink" href="#--bam_location_optional" title="Permanent link">&para;</a></h2>
<p>Path to directory holding all the alignment files for strains in the analysis. Defaults to <code>/vast/eande106/data/{species}/WI/alignments/</code></p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Remember to move your bam files output from <a href="https://github.com/andersenlab/alignment-nf">alignment-nf</a> to this location prior to running <code>wi-gatk</code>. In <strong>most</strong> cases, you will want to run <code>wi-gatk</code> on all samples, new and old combined.</p>
</div>
<h2 id="--species_optional">--species (optional)<a class="headerlink" href="#--species_optional" title="Permanent link">&para;</a></h2>
<p><strong>default</strong> = c_elegans</p>
<p>Options: c_elegans, c_briggsae, or c_tropicalis</p>
<h2 id="--project_optional">--project (optional)<a class="headerlink" href="#--project_optional" title="Permanent link">&para;</a></h2>
<p><strong>default</strong> = PRJNA13758</p>
<p>WormBase project ID for selected species. Choose from some examples <a href="https://github.com/AndersenLab/genomes-nf/blob/master/bin/project_species.tsv">here</a></p>
<h2 id="--ws_build_optional">--ws_build (optional)<a class="headerlink" href="#--ws_build_optional" title="Permanent link">&para;</a></h2>
<p><strong>default</strong> = WS283</p>
<p>WormBase version to use for reference genome.</p>
<h2 id="--reference_optional">--reference (optional)<a class="headerlink" href="#--reference_optional" title="Permanent link">&para;</a></h2>
<p>A fasta reference indexed with BWA. On Rockfish, the reference is available here:</p>
<pre><code>/vast/eande106/data/c_elegans/genomes/PRJNA13758/WS283/c_elegans.PRJNA13758.WS283.genome.fa.gz
</code></pre>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If running on Rockfish, instead of changing the <code>reference</code> parameter, opt to change the <code>species</code>, <code>project</code>, and <code>ws_build</code> for other species like c_briggsae (and then the reference will change automatically) </p>
</div>
<h2 id="--mito_name_optional">--mito_name (optional)<a class="headerlink" href="#--mito_name_optional" title="Permanent link">&para;</a></h2>
<p>Name of contig to skip het polarization. Might need to change for other species besides c_elegans if the mitochondria contig is named differently. Defaults to <code>MtDNA</code>.</p>
<h2 id="--output_optional">--output (optional)<a class="headerlink" href="#--output_optional" title="Permanent link">&para;</a></h2>
<p>A directory in which to output results. By default it will be <code>WI-YYYYMMDD</code> where YYYYMMDD is todays date.</p>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<p>The final output directory looks like this:</p>
<pre><code>
├── variation
│   ├── *.hard-filter.vcf.gz
│   ├── *.hard-filter.vcf.tbi
│   ├── *.hard-filter.stats.txt
│   ├── *.hard-filter.filter_stats.txt
│   ├── *.soft-filter.vcf.gz
│   ├── *.soft-filter.vcf.tbi
│   ├── *.soft-filter.stats.txt
│   └── *.soft-filter.filter_stats.txt
└── report
    ├── multiqc.html
    └── multiqc_data
        └── multiqc_*.json
   
</code></pre>
<h1 id="data_storage">Data Storage<a class="headerlink" href="#data_storage" title="Permanent link">&para;</a></h1>
<h2 id="cleanup">Cleanup<a class="headerlink" href="#cleanup" title="Permanent link">&para;</a></h2>
<p>The <code>hard-filter.vcf</code> is the input for both the <a href="https://github.com/AndersenLab/concordance-nf"><code>concordance-nf</code></a> pipeline and the <a href="https://github.com/AndersenLab/post-gatk-nf"><code>post-gatk-nf</code></a> pipeline. Once both pipelines have been completed successfully, the hard and soft filter vcf and index files (everything output in the <code>variation</code> folder) can be moved to <code>/vast/eande106/data/{species}/WI/variation/{date}/vcf</code>.</p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../pipeline-alignment/" class="btn btn-neutral float-left" title="alignment-nf"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../pipeline-concordance/" class="btn btn-neutral float-right" title="concordance-nf">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-alignment/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../pipeline-concordance/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/main.js"></script>
      <script src="../../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
