<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="shortcut icon" href="../../img/favicon.ico" />
    <title>nil-ril-nf - Andersen Lab Dry Guide</title>
    <link rel="stylesheet" href="../../css/theme.css" />
    <link rel="stylesheet" href="../../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "nil-ril-nf";
        var mkdocs_page_input_path = "pipelines/pipeline-nil-ril.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href="../.." class="icon icon-home"> Andersen Lab Dry Guide
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Home</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../..">Overview</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Knowledge Base</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/best_practices/">Best Practices</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/r/">R</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/shiny/">R Shiny</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/bash/">Command Line</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../knowledge_base/github/">Git and Github</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Rockfish</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/VAST/">Data on Rockfish</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../rockfish/rf-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quest</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-intro/">Introduction</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/b1059/">Data on b1059</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-conda/">Conda</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../quest/quest-nextflow/">Nextflow</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Nextflow Pipelines</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" >WI Sequencing pipelines</a>
    <ul>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" >Other pipelines</a>
    <ul class="current">
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                </li>
                <li class="toctree-l2 current"><a class="reference internal current" href="#">nil-ril-nf</a>
    <ul class="current">
    </ul>
                </li>
                <li class="toctree-l2"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../docker/">Docker</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../nf-GCPconfig/">GCP</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Other things</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/caendr/">CaeNDR</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/backup/">Backup</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sra/">SRA</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/cloud/">Cloud</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/sharing-globus/">Globus file sharing</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/writing-nextflow/">Writing Nextflow workflows</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../../other/labsite/">Andersen Labsite</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../..">Andersen Lab Dry Guide</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../.." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Nextflow Pipelines</li>
          <li class="breadcrumb-item">Other pipelines</li>
      <li class="breadcrumb-item active">nil-ril-nf</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/pipelines/pipeline-nil-ril.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="nil-ril-nf">nil-ril-nf<a class="headerlink" href="#nil-ril-nf" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#nil-ril-nf">nil-ril-nf</a></li>
<li><a href="#overview">Overview</a><ul>
<li><a href="#software_requirements">Software Requirements</a><ul>
<li><a href="#relevant_docker_images">Relevant Docker Images</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#usage">Usage</a><ul>
<li><a href="#testing_on_rockfish">Testing on Rockfish</a></li>
<li><a href="#running_on_rockfish">Running on Rockfish</a></li>
<li><a href="#running_the_pipeline_locally">Running the pipeline locally</a></li>
</ul>
</li>
<li><a href="#parameters">Parameters</a><ul>
<li><a href="#-profile">-profile</a></li>
<li><a href="#--debug">--debug</a></li>
<li><a href="#--fqs">--fqs</a></li>
<li><a href="#--vcf">--vcf</a></li>
<li><a href="#--reference">--reference</a></li>
<li><a href="#--a_--b_optional">--A, --B (optional)</a></li>
<li><a href="#--cores_optional">--cores (optional)</a></li>
<li><a href="#--ca_--cb_optional">--cA, --cB (optional)</a></li>
<li><a href="#--out_optional">--out (optional)</a></li>
<li><a href="#--relative_optional">--relative (optional)</a></li>
<li><a href="#--cross_obj_optional">--cross_obj (optional)</a></li>
<li><a href="#--tmpdir_optional">--tmpdir (optional)</a></li>
</ul>
</li>
<li><a href="#output">Output</a><ul>
<li><a href="#logtxt">log.txt</a></li>
<li><a href="#duplicates">duplicates/</a></li>
<li><a href="#fq">fq/</a></li>
<li><a href="#sm">SM/</a></li>
<li><a href="#hmm">hmm/</a></li>
<li><a href="#plots">plots/</a></li>
<li><a href="#sitelist">sitelist/</a></li>
</ul>
</li>
<li><a href="#organizing_final_data">Organizing final data</a></li>
<li><a href="#adding_nil_sequence_data_to_lab_website">Adding NIL sequence data to lab website</a></li>
</ul>
</div>
<p>The <code>nil-ril-nf</code> pipeline will align, call variants, and generate datasets for NIL and RIL sequence data. It runs a hidden-markov-model to fill in missing genotypes from low-coverage sequence data.</p>
<h1 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h1>
<pre><code class="language-bash">
    ███╗   ██╗██╗██╗      ██████╗ ██╗██╗      ███╗   ██╗███████╗
    ████╗  ██║██║██║      ██╔══██╗██║██║      ████╗  ██║██╔════╝
    ██╔██╗ ██║██║██║█████╗██████╔╝██║██║█████╗██╔██╗ ██║█████╗
    ██║╚██╗██║██║██║╚════╝██╔══██╗██║██║╚════╝██║╚██╗██║██╔══╝
    ██║ ╚████║██║███████╗ ██║  ██║██║███████╗ ██║ ╚████║██║
    ╚═╝  ╚═══╝╚═╝╚══════╝ ╚═╝  ╚═╝╚═╝╚══════╝ ╚═╝  ╚═══╝╚═╝


    parameters           description                    Set/Default
    ==========           ===========                    =======

    --debug              Set to 'true' to test          false
    --cores              Number of cores                4
    --A                  Parent A                       N2
    --B                  Parent B                       CB4856
    --cA                 Parent A color (for plots)     #0080FF
    --cB                 Parent B color (for plots)     #FF8000
    --out                Directory to output results    NIL-N2-CB4856-2017-09-27
    --fqs                fastq file (see help)          (required)
    --relative           use relative fastq prefix      true
    --reference          Reference Genome               (required)
    --vcf                VCF to fetch parents from      (required)
    --tmpdir             A temporary directory          tmp/


    The Set/Default column shows what the value is currently set to
    or would be set to if it is not specified (its default).


</code></pre>
<p><img alt="Overview" src="../../img/nil_ril.drawio.svg" /></p>
<ol>
<li><code>Alignment</code> - Performed using bwa-mem</li>
<li><code>Merge Bams</code> - Combines bam files aligned individually for each fastq-pair. <a href="http://lomereiter.github.io/sambamba/">Sambamba</a> is actually used in place of samtools, but it's a drop-in, faster replacement.</li>
<li><code>Bam Stats</code> - A variety of metrics are calculated for bams and combined into individual files for downstream analsyis.</li>
<li><code>Mark Duplicates</code> - Duplicate reads are marked using Picard.</li>
<li><code>Call Variants individual</code> - Variants are called for each strain inidividually first. This generates a sitelist which is used to identify all variant sites in the population.</li>
<li><code>Pull parental genotypes</code> - Pulls out parental genotypes from the given VCF. The list of genotypes is filtered for discordant calls (i.e. different genotypes). This is VCF is used to generate a sitelist for calling low-coverage bams and later is merged into the resulting VCF.</li>
<li><code>Call variants union</code> - Uses the sitelist from the previous step to call variants on low-coverage sequence data. The resulting VCF will have a lot of missing calls.</li>
<li><code>Merge VCF</code> - Merges in the parental VCF (which has been filtered only for variants with discordant calls).</li>
<li><code>Call HMM</code> - VCF-kit is run in various ways to infer the appropriate genotypes from the low-coverage sequence data.</li>
</ol>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Do not perform any pre-processing on NIL data. NIL-data is low-coverage by design and you want to retain as much sequence data (however poor) as possible.</p>
</div>
<h2 id="software_requirements">Software Requirements<a class="headerlink" href="#software_requirements" title="Permanent link">&para;</a></h2>
<ul>
<li>The latest update requires Nextflow version 23+. On Rockfish, you can access this version by loading the <code>nf23_env</code> conda environment prior to running the pipeline command:</li>
</ul>
<pre><code>module load python/anaconda
source activate /data/eande106/software/conda_envs/nf23_env
</code></pre>
<h3 id="relevant_docker_images">Relevant Docker Images<a class="headerlink" href="#relevant_docker_images" title="Permanent link">&para;</a></h3>
<p>The docker image used by the <code>nil-ril-nf</code> pipeline is the <code>nil-ril-nf</code> docker image:</p>
<p><a href="https://hub.docker.com/r/andersenlab/nil-ril-nf/">andersenlab/nil-ril-nf</a></p>
<p>The <strong>Dockerfile</strong> is stored in the root of the <code>nil-ril-nf</code> github repo and is automatically built on <a href="http://www.dockerhub.com">Dockerhub</a> whenever the repo is pushed.</p>
<h1 id="usage">Usage<a class="headerlink" href="#usage" title="Permanent link">&para;</a></h1>
<p><em>Note: if you are having issues running Nextflow or need reminders, check out the <a href="../../rockfish/rf-nextflow/">Nextflow</a> page.</em></p>
<h2 id="testing_on_rockfish">Testing on Rockfish<a class="headerlink" href="#testing_on_rockfish" title="Permanent link">&para;</a></h2>
<p><em>This command uses a test dataset</em></p>
<pre><code>nextflow run -latest andersenlab/nil-ril-nf --debug
</code></pre>
<h2 id="running_on_rockfish">Running on Rockfish<a class="headerlink" href="#running_on_rockfish" title="Permanent link">&para;</a></h2>
<p>You should run this in a screen or tmux session.</p>
<pre><code>nextflow run -latest andersenlab/nil-ril-nf -resume
</code></pre>
<h2 id="running_the_pipeline_locally">Running the pipeline locally<a class="headerlink" href="#running_the_pipeline_locally" title="Permanent link">&para;</a></h2>
<p>When running locally, the pipeline will run using the several different docker images. You must have docker installed. You will need to obtain a reference genome to run the alignment with as well. You can use the following command to obtain the reference:</p>
<pre><code class="language-bash">curl ftp://wormbase.org/pub/wormbase/releases/WS276/species/c_elegans/PRJNA13758/c_elegans.PRJNA13758.WS276.genomc.fa.gz &gt; c_elegans.PRJNA13758.WS276.fa.gz
</code></pre>
<p>You will also need to build a <code>bwa mem</code> index or copy it from Rockfish:</p>
<pre><code class="language-bash">for I in rpac amb ann pac bwt rbwt rsa sa; do
  scp &lt;JHED_ID&gt;@login.rockfish.jhu.edu:/vast/eande106/data/c_elegans/genomes/PRJNA13758/WS276/c_elegans.PRJNA13758.WS276.genomc.fa.gz.${I} ./
done
</code></pre>
<p>Run the pipeline locally with:</p>
<pre><code class="language-bash">nextflow run -latest andersenlab/nil-ril-nf -profile local -resume
</code></pre>
<h1 id="parameters">Parameters<a class="headerlink" href="#parameters" title="Permanent link">&para;</a></h1>
<h2 id="-profile">-profile<a class="headerlink" href="#-profile" title="Permanent link">&para;</a></h2>
<p>There are three configuration profiles for this pipeline.</p>
<ul>
<li><code>rockfish</code> - Used for running on Rockfish (default).</li>
<li><code>quest</code>    - Used for running on Quest.</li>
<li><code>local</code>    - Used for local development.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you forget to add a <code>-profile</code>, the <code>rockfish</code> profile will be chosen as default</p>
</div>
<h2 id="--debug">--debug<a class="headerlink" href="#--debug" title="Permanent link">&para;</a></h2>
<p>You should use <code>--debug</code> for testing/debugging purposes. This will run the debug test set (located in the <code>test_data</code> folder).</p>
<p>For example:</p>
<pre><code>nextflow run -latest andersenlab/nil-ril-nf --debug
</code></pre>
<h2 id="--fqs">--fqs<a class="headerlink" href="#--fqs" title="Permanent link">&para;</a></h2>
<p>In order to process NIL/RIL data, you need to move the sequence data to a folder and create a <code>fq_sheet.tsv</code>. This file defines the fastqs that should be processed. The fastq can be specified as <em>relative</em> or <em>absolute</em>. By default, they are expected to be relative to the execution directory. The FASTQ sheet details strain names, ids, library, and files. It should be tab-delimited and look like this (this shows relative paths to fastqs):</p>
<pre><code>NIL_01   NIL_01_ID    S16 &lt;relative_path_to_fq&gt;/NIL_01_1.fq.gz   &lt;relative_path_to_fq&gt;/NIL_01_2.fq.gz
NIL_02   NIL_02_ID    S1  &lt;relative_path_to_fq&gt;/NIL_02_1.fq.gz   &lt;relative_path_to_fq&gt;/NIL_02_2.fq.gz
</code></pre>
<p>Notice that the file does not include a header. The table with corresponding columns looks like this (this shows absolute paths to fastqs).</p>
<table>
<thead>
<tr>
<th style="text-align: left;">strain</th>
<th style="text-align: left;">fastq_pair_id</th>
<th style="text-align: left;">library</th>
<th style="text-align: left;">fastq-1-path</th>
<th style="text-align: left;">fastq-2-path</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">NIL_01</td>
<td style="text-align: left;">NIL_01_ID</td>
<td style="text-align: left;">S16</td>
<td style="text-align: left;">/<path_to_fq>/NIL_01_1.fq.gz</td>
<td style="text-align: left;">/<path_to_fq>/NIL_01_2.fq.gz</td>
</tr>
<tr>
<td style="text-align: left;">NIL_02</td>
<td style="text-align: left;">NIL_02_ID</td>
<td style="text-align: left;">S1</td>
<td style="text-align: left;">/<path_to_fq>/NIL_02_1.fq.gz</td>
<td style="text-align: left;">/<path_to_fq>/NIL_02_2.fq.gz</td>
</tr>
</tbody>
</table>
<p>The columns are detailed below:</p>
<ul>
<li><strong>strain</strong> - The name of the strain. If a strain was sequenced multiple times this file is used to identify that fact and merge those fastq-pairs together following alignment.</li>
<li><strong>fastq_pair_id</strong> - This must be unique identifier for all individual FASTQ pairs.</li>
<li><strong>library</strong> - A string identifying the DNA library. If you sequenced a strain from different library preps it can be beneficial when calling variants. The string can be arbitrary (e.g. LIB1) as well if only one library prep was used.</li>
<li><strong>fastq-1-path</strong> - The <strong>relative</strong> or <strong>absolute</strong> path of the first fastq.</li>
<li><strong>fastq-2-path</strong> - The <strong>relative</strong> or <strong>absolute</strong> path of the second fastq.</li>
</ul>
<p>This file needs to be placed along with the sequence data into a folder. The tree will look like this:</p>
<pre><code class="language-bash">NIL_SEQ_DATA/
├── NIL_01_1.fq.gz
├── NIL_01_2.fq.gz
├── NIL_02_1.fq.gz
├── NIL_02_2.fq.gz
└── fq_sheet.tsv
</code></pre>
<p>Set <code>--fqs</code> as <code>--fqs=/the/path/to/fq_sheet.tsv</code>.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Do not include the parental strains in the fq_sheet. If you re-sequenced the parent strains and want to include them in the analysis as a control, you need to rename the parent strains to avoid an error in merging the VCFs (i.e. N2 becomes N2-1).</p>
</div>
<h2 id="--vcf">--vcf<a class="headerlink" href="#--vcf" title="Permanent link">&para;</a></h2>
<p>Before you begin, you will need access to a VCF with high-coverage data from the parental strains. In general, this can be obtained using the latest release of the wild-isolate data which is usually located in the <code>/vast/eande106/data</code> analysis folder (on Rockfish). For example, the most recent <em>C. elegans</em> VCF could be found here:</p>
<pre><code>/vast/eande106/data/c_elegans/WI/variation/20210121/vcf/WI.20210121.hard-filter.isotype.vcf.gz
</code></pre>
<p>This is the <strong>hard-filtered</strong> VCF, meaning that poor quality variants have been stripped. Use hard-filtered VCFs for this pipeline.</p>
<p>Set the parental VCF as <code>--vcf=/the/path/to/WI.20210121.hard-filter.isotype.vcf.gz</code></p>
<h2 id="--reference">--reference<a class="headerlink" href="#--reference" title="Permanent link">&para;</a></h2>
<p>A fasta reference indexed with BWA. For example, the <em>C. elegans</em> reference could be found here:</p>
<pre><code>/vast/eande106/data/c_elegans/genomes/PRJNA13758/WS276/c_elegans.PRJNA13758.WS276.genome.fa.gz
</code></pre>
<h2 id="--a_--b_optional">--A, --B (optional)<a class="headerlink" href="#--a_--b_optional" title="Permanent link">&para;</a></h2>
<p>Two parental strains must be provided. By default these are N2 and CB4856. The parental strains provided <strong>must</strong> be present in the VCF provided. Their genotypes are pulled from that VCF and used to generate the HMM. See below for more details.</p>
<h2 id="--cores_optional">--cores (optional)<a class="headerlink" href="#--cores_optional" title="Permanent link">&para;</a></h2>
<p>The number of cores to use during alignments and variant calling. Default is 4.</p>
<h2 id="--ca_--cb_optional">--cA, --cB (optional)<a class="headerlink" href="#--ca_--cb_optional" title="Permanent link">&para;</a></h2>
<p>The color to use for parental strain A and B on plots. Default is orange and blue.</p>
<h2 id="--out_optional">--out (optional)<a class="headerlink" href="#--out_optional" title="Permanent link">&para;</a></h2>
<p>A directory in which to output results. By default it will be <code>NIL-A-B-YYYY-MM-DD</code> where A and be are the parental strains.</p>
<h2 id="--relative_optional">--relative (optional)<a class="headerlink" href="#--relative_optional" title="Permanent link">&para;</a></h2>
<p>If you want to specify fastqs using an absolute path use <code>--relative=false</code>. Set to <code>true</code> by default. </p>
<h2 id="--cross_obj_optional">--cross_obj (optional)<a class="headerlink" href="#--cross_obj_optional" title="Permanent link">&para;</a></h2>
<p>If you are running a set of RILs, you might want to add the <code>--cross_obj true</code> parameter. When <code>true</code>, the pipeline will run an additional step to pair down the total genetic variants to only the informative variants and output a smaller genotype matrix to input directly into a new cross object. An example of how to generate a cross object can be found in the <code>bin</code>. There is no need to run this option for NIL data.</p>
<h2 id="--tmpdir_optional">--tmpdir (optional)<a class="headerlink" href="#--tmpdir_optional" title="Permanent link">&para;</a></h2>
<p>A directory for storing temporary data.</p>
<h1 id="output">Output<a class="headerlink" href="#output" title="Permanent link">&para;</a></h1>
<p>The final output directory looks like this:</p>
<pre><code class="language-bash">.
├── log.txt
├── fq
│   ├── fq_bam_idxstats.tsv
│   ├── fq_bam_stats.tsv
│   ├── fq_coverage.full.tsv
│   └── fq_coverage.tsv
├── SM
│   ├── SM_bam_idxstats.tsv
│   ├── SM_bam_stats.tsv
│   ├── SM_coverage.full.tsv
│   ├── SM_union_vcfs.txt
│   └── SM_coverage.tsv
├── hmm
│   ├── gt_hmm.(png/svg)
│   ├── gt_hmm.tsv
│   ├── gt_hmm_fill.tsv
│   ├── NIL.filtered.stats.txt
│   ├── NIL.filtered.vcf.gz
│   ├── NIL.filtered.vcf.gz.csi
│   ├── NIL.hmm.vcf.gz
│   ├── NIL.hmm.vcf.gz.csi
│   └── gt_hmm_genotypes.tsv
├── bam
│   └── &lt;BAMS + indices&gt;
├── duplicates
│   └── bam_duplicates.tsv
└─ sitelist
    ├── N2.CB4856.sitelist.[tsv/vcf].gz
    └── N2.CB4856.sitelist.[tsv/vcf].gz.[tbi/csi]
</code></pre>
<h3 id="logtxt">log.txt<a class="headerlink" href="#logtxt" title="Permanent link">&para;</a></h3>
<p>A summary of the nextflow run.</p>
<h3 id="duplicates">duplicates/<a class="headerlink" href="#duplicates" title="Permanent link">&para;</a></h3>
<p><strong>bam_duplicates.tsv</strong> - A summary of duplicate reads from aligned bams.</p>
<h3 id="fq">fq/<a class="headerlink" href="#fq" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>fq_bam_idxstats.tsv</strong> - A summary of mapped and unmapped reads by fastq pair.</li>
<li><strong>fq_bam_stats.tsv</strong> - BAM summary by fastq pair.</li>
<li><strong>fq_coverage.full.tsv</strong> - Coverage summary by chromosome</li>
<li><strong>fq_coverage.tsv</strong> - Simple coverage file by fastq</li>
</ul>
<h3 id="sm">SM/<a class="headerlink" href="#sm" title="Permanent link">&para;</a></h3>
<p>If you have multiple fastq pairs per sample, their alignments will be combined into a strain or sample-level BAM and the results will be output to this directory.</p>
<ul>
<li><strong>SM_bam_idxstats.tsv</strong> - A summary of mapped and unmapped reads by sample.</li>
<li><strong>SM_bam_stats.tsv</strong> - BAM summary at the sample level</li>
<li><strong>SM_coverage.full.tsv</strong> - Coverage at the sample level</li>
<li><strong>SM_coverage.tsv</strong> - Simple coverage at the sample level.</li>
<li><strong>SM_union_vcfs.txt</strong> - A list of VCFs that were merged to generate RIL.filter.vcf.gz</li>
</ul>
<h3 id="hmm">hmm/<a class="headerlink" href="#hmm" title="Permanent link">&para;</a></h3>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p><strong>gt_hmm_fill.tsv</strong> is for visualization purposes only. To determine breakpoints you should use <strong>gt_hmm.tsv</strong>.</p>
</div>
<p><img alt="hmm_fill" src="http://vcf-kit.readthedocs.io/en/latest/hmm_opts.png" /></p>
<p>The <code>--infill</code> and <code>--endfill</code> options are applied to the <strong>gt_hmm_fill.tsv</strong> file. You need to be cautious when examining this data as it is <strong>generated primarily for visualization purposes</strong>.</p>
<ul>
<li><strong>gt_hmm.(png/svg)</strong> - Haplotype plot <strong>using</strong> <code>--infill</code> and <code>--endfill</code>.</li>
<li><strong>gt_hmm_fill.tsv</strong> - Same as above, but using <code>--infill</code> and <code>--endfill</code> with VCF-Kit. For more information, see <a href="http://vcf-kit.readthedocs.io/en/latest/">VCF-Kit Documentation</a>. This file is used to generate the plots.</li>
<li><strong>gt_hmm.tsv</strong> - Haplotypes defined by region with associated information. Does <strong>not</strong> use <code>--infill</code> and <code>--endfill</code></li>
<li><strong>gt_hmm_genotypes.tsv</strong> - Long form genotypes file.</li>
<li><strong>NIL/RIL.filtered.vcf.gz</strong> - A VCF genotypes including the NILs and parental genotypes.</li>
<li><strong>NIL/RIL.filtered.stats.txt</strong> - Summary of filtered genotypes. Generated by <code>bcftools stats NIL.filtered.vcf.gz</code></li>
<li><strong>NIL/RIL.hmm.vcf.gz</strong> - The NIL/RIL VCF as output by VCF-Kit; HMM applied to determine genotypes.</li>
</ul>
<h3 id="plots">plots/<a class="headerlink" href="#plots" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>coverage_comparison.png</strong> - Compares FASTQ and Sample-level coverage. Note that coverage is not simply cumulative. Only uniquely mapped reads count towards coverage, so it is possible that the sample-level coverage will not equal to the cumulative sum of the coverages of individual FASTQ pairs.</li>
<li><strong>duplicates.(png/pdf)</strong> - Coverage vs. percent duplicated.</li>
<li><strong>unmapped_reads.png</strong> - Coverage vs. unmapped read percent.</li>
</ul>
<h3 id="sitelist">sitelist/<a class="headerlink" href="#sitelist" title="Permanent link">&para;</a></h3>
<ul>
<li><code>&lt;A&gt;.&lt;B&gt;.sitelist.tsv.gz[+.tbi]</code> - A tabix-indexed list of sites found to be different between both parental strains.</li>
<li><code>&lt;A&gt;.&lt;B&gt;.sitelist.vcf.gz[+.tbi]</code> - A vcf of sites found to be different between both parental strains.</li>
</ul>
<h1 id="organizing_final_data">Organizing final data<a class="headerlink" href="#organizing_final_data" title="Permanent link">&para;</a></h1>
<p>After the run is complete and you are satisfied with the results, follow these steps to ensure correct data storage on QUEST:</p>
<ol>
<li>Move the raw fastq files to <code>/vast/eande106/data/{species}/{NIL or RIL}/fastq/</code>. You might want to use <code>mv -i</code> to ensure no files are overwritten.</li>
<li>Move the BAM files from the output folder to <code>/vast/eande106/data/{species}/{NIL or RIL}/alignments/</code>. You might want to use <code>mv -i</code> to ensure no files are overwritten.</li>
<li>Delete the now empty <code>bam</code> folder in the output directory.</li>
<li>Move the sample sheet generated for analysis into the output directory.</li>
<li>Make sure the output directory follows the default naming structure that is informative about the analysis (i.e. <code>NIL-20200322-N2-CB4856</code> (if NIL/RIL analysis is performed for another lab, consider adding a <code>-{LabName}</code> like <code>-Baugh</code> to the end of the folder name)).</li>
<li>Move the entire output folder to <code>/vast/eande106/data/{species}/{NIL or RIL}/variation</code>.</li>
</ol>
<h1 id="adding_nil_sequence_data_to_lab_website">Adding NIL sequence data to lab website<a class="headerlink" href="#adding_nil_sequence_data_to_lab_website" title="Permanent link">&para;</a></h1>
<p>If your sequencing was N2-CB4856 NILs (and maybe other <em>C. elegans</em> NILs as well...?) you probably want to add this sequencing data to the lab website to be accessed by everyone when looking for NIL genotypes.</p>
<p>Check out <a href="../adding-seq-data/">this page</a> for instructions on how to do that. Once done, you should be able to view your NILs on the <a href="https://andersen-lab.shinyapps.io/nil-browser/">NIL browser shiny app</a>.</p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../pipeline-cellprofiler/" class="btn btn-neutral float-left" title="Image Analysis"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../pipeline-genomes-nf/" class="btn btn-neutral float-right" title="genomes-nf">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-cellprofiler/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../pipeline-genomes-nf/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "../..";</script>
    <script src="../../js/theme_extra.js"></script>
    <script src="../../js/theme.js"></script>
      <script src="../../search/main.js"></script>
      <script src="../../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
