<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>CeNDR - Andersen Lab Dry Guide</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  <link href="../css/version-select.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "CeNDR";
    var mkdocs_page_input_path = "cendr.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Andersen Lab Dry Guide</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Knowledge Base</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../best_practices/">Best Practices</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../r/">R</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../shiny/">R Shiny</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../bash/">Command Line</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../github/">Git and Github</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Quest</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-intro/">Introduction</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../b1059/">Data on b1059</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-nextflow/">Nextflow</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../quest-conda/">Conda</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-docker/">Docker</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-GCPconfig/">GCP</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">WI Sequencing pipelines</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-overview/">Overview</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-trimming/">trim-fq-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-alignment/">alignment-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-wiGATK/">wi-gatk</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-concordance/">concordance-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-postGATK/">post-gatk-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-impute/">imputation</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-annotation-nf/">annotation-nf</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Other pipelines</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-nemascan/">NemaScan</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-cellprofiler/">Image Analysis</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-nil-ril/">nil-ril-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-genomes-nf/">genomes-nf</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../pipeline-cegwas/">cegwas2-nf</a>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Other things</span></p>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">CeNDR</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#setting_up_cendr">Setting up CeNDR</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#clone_the_repo">Clone the repo</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#setup_a_python_environment">Setup a python environment</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#download_and_install_the_gcloud-sdk">Download and install the gcloud-sdk</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#create_a_cendr_gcloud_configuration">Create a cendr gcloud configuration</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#install_direnv">Install direnv</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#test_flask">Test flask</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#setup_credentials">Setup Credentials</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#load_the_database">Load the database</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#test_the_site">Test the site</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#creating_a_new_release">Creating a new release</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#uploading_bams">Uploading BAMs</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#uploading_release_data">Uploading Release Data</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#bump_the_cendr_version_number_and_change-log">Bump the CeNDR version number and change-log</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#adding_the_release_to_the_cendr_website">Adding the release to the CeNDR website</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#adding_isotype_images">Adding isotype images</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#update_the_current_variant_datasets">Update the current variant datasets.</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#updating_publications">Updating Publications</a>
    </li>
    </ul>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../backup/">Backup</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../cloud/">Cloud</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../sharing-globus/">Globus file sharing</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../labsite/">Andersen Labsite</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Andersen Lab Dry Guide</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Other things &raquo;</li>
        
      
    
    <li>CeNDR</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/Andersenlab/dry-guide/edit/master/docs/cendr.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h1 id="cendr">CeNDR<a class="headerlink" href="#cendr" title="Permanent link">&para;</a></h1>
<div class="toc">
<ul>
<li><a href="#cendr">CeNDR</a><ul>
<li><a href="#setting_up_cendr">Setting up CeNDR</a><ul>
<li><a href="#clone_the_repo">Clone the repo</a></li>
<li><a href="#setup_a_python_environment">Setup a python environment</a></li>
<li><a href="#download_and_install_the_gcloud-sdk">Download and install the gcloud-sdk</a></li>
<li><a href="#create_a_cendr_gcloud_configuration">Create a cendr gcloud configuration</a></li>
<li><a href="#install_direnv">Install direnv</a></li>
<li><a href="#test_flask">Test flask</a></li>
<li><a href="#setup_credentials">Setup Credentials</a></li>
<li><a href="#load_the_database">Load the database</a></li>
<li><a href="#test_the_site">Test the site</a></li>
</ul>
</li>
<li><a href="#creating_a_new_release">Creating a new release</a><ul>
<li><a href="#uploading_bams">Uploading BAMs</a></li>
<li><a href="#uploading_release_data">Uploading Release Data</a></li>
<li><a href="#bump_the_cendr_version_number_and_change-log">Bump the CeNDR version number and change-log</a></li>
<li><a href="#adding_the_release_to_the_cendr_website">Adding the release to the CeNDR website</a></li>
<li><a href="#adding_isotype_images">Adding isotype images</a></li>
<li><a href="#update_the_current_variant_datasets">Update the current variant datasets.</a></li>
</ul>
</li>
<li><a href="#updating_publications">Updating Publications</a></li>
</ul>
</li>
</ul>
</div>
<h2 id="setting_up_cendr">Setting up CeNDR<a class="headerlink" href="#setting_up_cendr" title="Permanent link">&para;</a></h2>
<h3 id="clone_the_repo">Clone the repo<a class="headerlink" href="#clone_the_repo" title="Permanent link">&para;</a></h3>
<p>Clone the repo first.</p>
<pre><code>git clone http://www.github.com/andersenlab/cendr
</code></pre>
<p><strong>Switch to the development branch</strong></p>
<pre><code>git checkout --track origin/development
</code></pre>
<h3 id="setup_a_python_environment">Setup a python environment<a class="headerlink" href="#setup_a_python_environment" title="Permanent link">&para;</a></h3>
<p>Use <a href="https://docs.conda.io/en/latest/miniconda.html">miniconda</a> as it will make your life much easier.</p>
<p>The conda environment has been specified in the <code>env.yaml</code> file, and can be installed using:</p>
<pre><code class="language-bash">conda env create -f env.yaml
</code></pre>
<h3 id="download_and_install_the_gcloud-sdk">Download and install the gcloud-sdk<a class="headerlink" href="#download_and_install_the_gcloud-sdk" title="Permanent link">&para;</a></h3>
<p>Install the <a href="https://cloud.google.com/sdk/downloads">gcloud-sdk</a></p>
<h3 id="create_a_cendr_gcloud_configuration">Create a cendr gcloud configuration<a class="headerlink" href="#create_a_cendr_gcloud_configuration" title="Permanent link">&para;</a></h3>
<pre><code>gcloud config configurations create cendr
</code></pre>
<h3 id="install_direnv">Install direnv<a class="headerlink" href="#install_direnv" title="Permanent link">&para;</a></h3>
<p><a href="https://direnv.net/">direnv</a> allows you to load a configuration file when you enter the development directory. Please read about how it works. CeNDR uses a <code>.envrc</code> file within the repo
to set up the appropriate environmental variables.</p>
<p>Once direnv is installed you can run <code>direnv allow</code> within the CeNDR repo:</p>
<pre><code class="language-bash">direnv allow
</code></pre>
<h3 id="test_flask">Test flask<a class="headerlink" href="#test_flask" title="Permanent link">&para;</a></h3>
<p>With direnv enabled, you are nearly able to run the site locally.</p>
<p>Run <code>flask</code>, and you should see the following:</p>
<pre><code>&gt; flask
Usage: flask [OPTIONS] COMMAND [ARGS]...

  A general utility script for Flask applications.

  Provides commands from Flask, extensions, and the application. Loads the
  application defined in the FLASK_APP environment variable, or from a
  wsgi.py file. Setting the FLASK_ENV environment variable to 'development'
  will enable debug mode.

    $ export FLASK_APP=hello.py
    $ export FLASK_ENV=development
    $ flask run

Options:
  --version  Show the flask version
  --help     Show this message and exit.

Commands:
  decrypt_credentials  Decrypt credentials
  download_db          Download the database (used in docker...
  initdb               Initialize the database
  routes               Show the routes for the app.
  run                  Run a development server.
  shell                Runs a shell in the app context.
  update_credentials   Update credentials
  update_strains       Updates the strain table of the database

</code></pre>
<p>If you do not see the full set of commands there - something is broken.</p>
<h3 id="setup_credentials">Setup Credentials<a class="headerlink" href="#setup_credentials" title="Permanent link">&para;</a></h3>
<ol>
<li>Authenticate with gcloud.</li>
<li>Run the following command:</li>
</ol>
<pre><code class="language-bash">mkdir -p env_config
flask decrypt_credentials
</code></pre>
<p>This will create a directory with the site credentials (<code>env_config</code>). Keep these secret.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p><strong>DO NOT COMMIT THESE CREDENTIALS TO GITHUB</strong>!!!</p>
</div>
<h3 id="load_the_database">Load the database<a class="headerlink" href="#load_the_database" title="Permanent link">&para;</a></h3>
<p>The site uses an SQLite database that can be setup by running:</p>
<pre><code>flask download_db
</code></pre>
<p>This will update the SQLite database used by CeNDR (<code>base/cendr.db</code>). The tables are:</p>
<ul>
<li><code>homologs</code> - A table of homologs+orthologs.</li>
<li><code>strain</code> - Strain info pulled from the google spreadsheet <strong>C. elegans WI Strain Info</strong>.</li>
<li><code>wormbase_gene</code> - Summarizes gene information; Broken into component parts (e.g. exons, introns etc.).</li>
<li><code>wormbase_gene_summary</code> - Summarizes gene information. One line per gene.</li>
<li><code>metadata</code> - tracks how data was obtained. When. Where. etc.</li>
</ul>
<h3 id="test_the_site">Test the site<a class="headerlink" href="#test_the_site" title="Permanent link">&para;</a></h3>
<p>You can at this point test the site locally by running:</p>
<pre><code class="language-bash">flask run
</code></pre>
<p>Be sure you have direnv. Otherwise you should source the .envrc file prior to running:</p>
<pre><code class="language-bash">source .envrc
flask run
</code></pre>
<hr />
<h2 id="creating_a_new_release">Creating a new release<a class="headerlink" href="#creating_a_new_release" title="Permanent link">&para;</a></h2>
<p>Before a new release is possible, you must have first completed the following tasks:</p>
<p><strong>See <a href="../adding-seq-data/">Add new sequence data for further details</a></strong>.</p>
<ol>
<li>Add new wild isolate sequence data, and process with the <code>trimmomatic-nf</code> pipeline.</li>
<li>Identified new isotypes using the <code>concordance-nf</code></li>
<li>Updated the <code>C. elegans WI Strain Info</code> spreadsheet, adding in new isotypes. </li>
<li>Update the release column to reflect the release data in the <code>C. elegans WI Strain Info</code> spreadsheet</li>
<li>Run and process sequence data with the <code>wi-nf</code> pipeline.</li>
</ol>
<p>Pushing a new release requires a series of steps described below.</p>
<h3 id="uploading_bams">Uploading BAMs<a class="headerlink" href="#uploading_bams" title="Permanent link">&para;</a></h3>
<p>You will need AWS credentials to upload BAMs to Amazon S3. These are available in the secret credentials location. </p>
<pre><code>pip install aws-shell
aws configure # Use s3 user credentials
</code></pre>
<p>Once configured, navigate to the BAM location on b1059.</p>
<pre><code>cd /projects/b1059/data/alignments/WI/isotype
# CD to bams folder...
aws s3 sync . s3://elegansvariation.org/bam
</code></pre>
<p>Run this command in screen to ensure that it completes (it's going to take a while)</p>
<h3 id="uploading_release_data">Uploading Release Data<a class="headerlink" href="#uploading_release_data" title="Permanent link">&para;</a></h3>
<p>When you run the <code>wi-nf</code> pipeline it will create a folder with the format <code>WI-YYYYMMDD</code>. These data are output in a format that CeNDR can read as a release. You must upload the <code>WI-YYYYMMDD</code> folder to google storage with a command that looks like this:</p>
<pre><code># First cd to the path of the results folder (WI-YYYYMMDD) from the `wi-nf` pipeline.
gsutil rsync . gs://elegansvariation.org/releases/YYYYMMDD/
</code></pre>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>Use rsync to copy the files up to google storage. Note that the <code>WI-</code> prefix has been dropped from the <code>YYYYMMDD</code> declaration.</p>
</div>
<h3 id="bump_the_cendr_version_number_and_change-log">Bump the CeNDR version number and change-log<a class="headerlink" href="#bump_the_cendr_version_number_and_change-log" title="Permanent link">&para;</a></h3>
<p>Because we are creating a new data release, we need to "bump" or move up the CeNDR version. The CeNDR version number is specified in a file at the base of the repo: <code>travis.yml</code>. Modify this line:</p>
<pre><code>- export VERSION_NUM=1-2-8
</code></pre>
<p>And increase the version number by 1 (e.g. 1-2-9).</p>
<p>You should also update the change log and/or add a news item. The change-log is a markdown file located at <code>base/static/content/help/Change-Log.md</code>; News items are located at <code>base/static/news/</code>. Look at existing content to get an idea of how to add new items. It is fairly straightforward. You should be able to see changes on the test site.</p>
<h3 id="adding_the_release_to_the_cendr_website">Adding the release to the CeNDR website<a class="headerlink" href="#adding_the_release_to_the_cendr_website" title="Permanent link">&para;</a></h3>
<p>After the site is loaded, the BAMs and release data are up, and the database is updated, you need to modify the file <code>base/constants.py</code> to add the new release. The date must match the date of the release that was uploaded. Add your release with the appropriate 
date and the annotation database used (e.g. <code>("YYYYMMDD", "Annotation Database")</code>).</p>
<pre><code>RELEASES = [(&quot;20180413&quot;, &quot;WS263&quot;),
            (&quot;20170531&quot;, &quot;WS258&quot;),
            (&quot;20160408&quot;, &quot;WS245&quot;)]
</code></pre>
<p>Commit your changes to the <strong>development branch</strong> of CeNDR and push them to github. Once pushed, travis-ci will build the app and deploy it to a test branch. Use the google app engine interface to identify and test the app.</p>
<p>If everything looks good open a pull request bringing the changes on the development branch to the master branch. Again, travis-ci will launch the new site.</p>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>You need to shut down development instances and older versions of the site on the google-app engine interface once you are done testing/deploying new instances to prevent us from incurring charges for those running instances.</p>
</div>
<h3 id="adding_isotype_images">Adding isotype images<a class="headerlink" href="#adding_isotype_images" title="Permanent link">&para;</a></h3>
<p>Isolation photos are initially prepared on dropbox and are located in the folder here:</p>
<pre><code>~/Dropbox/Andersenlab/Reagents/WormReagents/isolation_photos/c_elegans
</code></pre>
<p>Each file should be named using the isotype name and the strain name strain name in the following format:</p>
<pre><code>&lt;isotype&gt;_&lt;strain&gt;.jpg
</code></pre>
<p>Then you will use <strong><a href="https://www.imagemagick.org/">imagemagick</a></strong> (a commandline-based utility) to scale the images down to 1000 pixels (width) and generate a 150px thumbnail.</p>
<pre><code>for img in `ls *.jpg`; do
    convert ${img} -density 300 -resize 1000 ${img}
    convert ${img} -density 300 -resize 150  ${img/.jpg/}.thumb.jpg
done;
</code></pre>
<p>Once you have generated the images you can upload them to google storage. They should be uploaded to the following location:</p>
<pre><code>gs://elegansvariation.org/photos/isolation
</code></pre>
<p>You can drag/drop the photos using the web-based browser or use <strong>gsutil</strong>:</p>
<pre><code># First cd to the appropriate directory
# cd ~/Dropbox/Andersenlab/Reagents/WormReagents/isolation_photos/c_elegans

gsutil rsync -x &quot;.DS_Store&quot; . gs://elegansvariation.org/photos/isolation
</code></pre>
<p>The script for processing files is located in the dropbox folder and is called 'process_images.sh'. It's also here:</p>
<pre><code>for img in `ls *.jpg`; do
    convert ${img} -density 300 -resize 1000 ${img}
    convert ${img} -density 300 -resize 150  ${img/.jpg/}.thumb.jpg
done;

# Copy using rsync; Skip .DS_Store files.
gsutil rsync -x &quot;.DS_Store&quot; . gs://elegansvariation.org/photos/isolation
</code></pre>
<h3 id="update_the_current_variant_datasets">Update the <code>current</code> variant datasets.<a class="headerlink" href="#update_the_current_variant_datasets" title="Permanent link">&para;</a></h3>
<p>The <code>current</code> folder located in <code>gs://elegansvariation.org/releases</code> contains the latest variant datasets and is used by WormBase to display natural variation data. Once you've completed a new release, update the files in this folder <code>gs://elegansvariation.org/releases/current</code> folder. </p>
<h2 id="updating_publications">Updating Publications<a class="headerlink" href="#updating_publications" title="Permanent link">&para;</a></h2>
<p>The publications page (<code>/about/publications</code>) is generated using a google spreadsheet. The spreadsheet can be accessed <a href="https://docs.google.com/spreadsheets/d/1ghJG6E_9YPsHu0H3C9s_yg_-EAjTUYBbO15c3RuePIs/edit">here</a>. You can request access to edit the spreadsheet by visiting that link.</p>
<p>The last row of the spreadsheet contains a function that can fetch publication data from Pubmed using its API. Simply fill in column A with the PMID (Pubmed Identifier), and the publication data will be fetched.</p>
<p>Once you have retrieved the latest pubmed data, create a new row and copy/paste the values for any new publications so they are not fetched from the Pubmed API.</p>
<p>Alternatively, you can fill in the details for a publication manually. In either case, any details added should be double checked. Changes should be instant, but there may be some dely on the CeNDR website.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../backup/" class="btn btn-neutral float-right" title="Backup">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../pipeline-cegwas/" class="btn btn-neutral" title="cegwas2-nf"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/Andersenlab/dry-guide/" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../pipeline-cegwas/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../backup/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
      <script src="../js/version-select.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
